---
title: "HTS DGE depression"
author:
  - name: Maya Śliwa
  - name: Igor Mokrytskyi
  - name: Jan Rysiak
date: 2026-10-07
date-format: "d MMM yyyy"
affiliations:
  - id: JU
    name: Jagiellonian University
format: 
  html:
    df-print: kable
    tidy: false
    embed-resources: false
    theme: lumen
    toc: true
    toc-location: left
    toc-depth: 6
    toc-expand: 6
    code-tools:
      source: true
      toggle: true
    code-fold: false
    code-summary: "Show the code"
    code-overflow: wrap
    page-layout: full
execute:
  eval: false
  cache: true
editor: visual
---

```{r}
#Load Packages
library(GGally)
library(DESeq2)
library(dplyr)
library(GEOquery)
library(tidyverse)
library(ggrepel)
library(apeglm)
library(data.table)
library(ggplot2) 
library(patchwork)  # for combining plots
library(org.Hs.eg.db)
library(AnnotationDbi)
```

# Download, filter and prepare data

## Download data

```{bash}
#Metadate
#wget --directory-prefix=data https://ftp.ncbi.nlm.nih.gov/geo/series/GSE260nnn/GSE260603/matrix/GSE260603_series_matrix.txt.gz

#Rawcounts for analyzis
#wget --directory-prefix=data https://ftp.ncbi.nlm.nih.gov/geo/series/GSE260nnn/GSE260603/suppl/GSE260603%5Fraw%5Fcounts.tsv.gz

#Normalize data
#wget --directory-prefix=data https://ftp.ncbi.nlm.nih.gov/geo/series/GSE260nnn/GSE260603/suppl/GSE260603%5Fnormalized%5Fdata.tsv.gz
```

## Prepare metadata

```{r}
#Read metadata and inspect
gse <- getGEO(filename="data/GSE260603_series_matrix.txt.gz", GSEMatrix=FALSE)

metadata <- pData(gse)

colnames(metadata)
```

```{r}
#Rename columns and extract nedeed columns
#Set new name
old_names <- c("age:ch1", "Sex:ch1", "bmi:ch1", "characteristics_ch1.4")
new_names <- c("age", "gender", "bmi", "affected")

#Rename
colnames(metadata)[match(old_names, colnames(metadata))] <- new_names
```

```{r}
#Deal with brake row. For 'GSM8120701' there is some ERROR in data. 
if (metadata['GSM8120701', 13] == 	"status: affected"){
  metadata['GSM8120701', 14:20] <- metadata['GSM8120701', 13:19] 
  metadata['GSM8120701', 13]<-NaN
  print(metadata[c('GSM8120701', 'GSM8120825', 'GSM8120638'), 13:20])
}
```

```{r}
#Filter, check and "civilize"  metadata
#Chose columns
metadata_filt<-metadata[c('title', 'age', 'gender', 'bmi','affected')]

#New rownems
metadata_filt$title <- gsub(".*(sample_[0-9]+).*", "\\1", metadata_filt$title)

#Drop rows with "status: affected; for replication analysis"
metadata_filt<-subset(metadata_filt, affected!="status: affected; for replication analysis")

#Chcnage afected
metadata_filt$affected <- gsub("status: (.*)", "\\1", metadata_filt$affected)

#Set new rownames
rownames(metadata_filt) <- metadata_filt$title
metadata_filt<-subset(metadata_filt, select=-(title))


class(metadata_filt$bmi)

# Convert BMI and age to numeric
metadata_filt$bmi <- as.numeric(metadata_filt$bmi)
metadata_filt$age <- as.numeric(as.character(metadata_filt$age))

metadata_filt
```

```{r}
#Visual check of metadata
bmi <- ggplot(metadata_filt, aes(x=bmi)) + 
  geom_histogram()
age <- ggplot(metadata_filt, aes(x=age)) + 
  geom_histogram()
(bmi|age)
```

```{r}
#Count male and female
summary_table <- metadata_filt %>%
  group_by(affected, gender) %>%
  summarise(n = n(), .groups = "drop")

summary_table

# Plot 
ggplot(summary_table, aes(x = affected, y = n, fill = gender)) +
  geom_bar(stat = "identity", position = "dodge") +
  theme_minimal() +
  labs(
    title = "Count of male and female in groups",
    x = "Group",
    y = "Liczba próbek",
    fill = "Gender"
  )

```

## Read and filter count matrix

```{r}
#Get count matrix
counts <- data.table::fread("data/GSE260603_raw_counts.tsv.gz")
```

```{r}
#Convert to array 
gene_ids <- counts[[1]]
count_matrix_raw <- as.matrix(counts[, -1])
rownames(count_matrix_raw) <- gene_ids
class(count_matrix_raw)
```

```{r}
#Filter out gene on chromosome Y
#Map Gene to chromosome
chr_map <- AnnotationDbi::select(
  org.Hs.eg.db,
  keys = rownames(count_matrix_raw),
  keytype = "ENSEMBL",
  columns = "CHR"
)

# Filter out Y
genes_Y <- chr_map$ENSEMBL[chr_map$CHR == "Y"]
genes_Y<-unique(genes_Y)

# Filter out genes form count matrix
count_matrix <- count_matrix_raw[
  !rownames(count_matrix_raw) %in% genes_Y,
]

```

## Read and filter Normalized data

```{r}
data_normalized <- read_tsv(file = "data/GSE260603_normalized_data.tsv")
data_normalized 
```

```{r}
# Transpose
df_t <- t(data_normalized)

# Convert to data frame
df_t <- as.data.frame(df_t)
colnames(df_t) <- as.character(unlist(df_t[1,]))
df_t<-df_t[-1,]
df_t
```

```{r}
expr <- data_normalized %>%
  as.data.frame()

rownames(expr) <- expr$Geneid
expr$Geneid <- NULL

expr <- as.matrix(expr)
storage.mode(expr) <- "numeric"
pca <- prcomp(t(expr), center = TRUE, scale. = TRUE)
```

```{r}
# data frame
pca_df_before <- as.data.frame(pca$x) %>%
  mutate(sample = rownames(.))
pca_df_before
```

## Inspect data

```{r}

# Combine PCA coordinates with metadata
pca_metadata_df <- cbind(pca_df_before, metadata_filt)

# pca gender
p1 <- ggplot(pca_metadata_df, aes(x = PC1, y = PC2, color = gender)) +
  geom_point(size = 3, alpha = 0.8) +
  labs(title = "Gender", color = "Gender") +
  theme_minimal(base_size = 12)

#pca affect
p2 <- ggplot(pca_metadata_df, aes(x = PC1, y = PC2, color = affected)) +
  geom_point(size = 3, alpha = 0.8) +
  labs(title = "Affected", color = "Affected") +
  theme_minimal(base_size = 12)

#pca bmi)
p3 <- ggplot(pca_metadata_df, aes(x = PC1, y = PC2, color = bmi)) +
  geom_point(size = 3, alpha = 0.8) +
  scale_color_gradient(low = "lightblue", high = "darkblue") +
  labs(title = "BMI", color = "BMI") +
  theme_minimal(base_size = 12)

# pca age
p4 <- ggplot(pca_metadata_df, aes(x = PC1, y = PC2, color = age)) +
  geom_point(size = 3, alpha = 0.8) +
  scale_color_gradient(low = "lightgreen", high = "darkgreen") +
  labs(title = "Age", color = "Age") +
  theme_minimal(base_size = 12)

# Combine 4 plots into 2x2 grid
(p1 | p2) / (p3 | p4)

```

### Checkpoint

```{r}

#Save raw data
save(metadata, counts, file = "RData/input_raw.RData")

#SAVE DATA load for dds 
save(metadata_filt, count_matrix, file = "RData/input_for_DGE_all.RData")
#load("RData/input_for_DGE_all.RData")
```

# DGE DESeq2

```{r}
#Perform DGE
dds <- DESeqDataSetFromMatrix(
  countData = count_matrix,
  colData = metadata_filt,
  design = ~ affected + gender
)

dds <- dds[rowSums(counts(dds)) > 10, ]
dds <- DESeq(dds)
```

```{r}
#Checkpoint
save(dds, file="RData/dds_all.Rdata")
#dds<-load("RData/dds_all.Rdata")
```

![]()

```{r}
#Set pval and logfc
pval=0.01
logfc=1
```

```{r}
#Normalize data
norm_counts <- DESeq2::counts(dds, normalized = T)

# Get result
res <- results(dds)

#Convert to data frame
dge_res_df <- as.data.frame(res)

#Drop row with Na in padj
dge_res_df <-drop_na(dge_res_df)

# Add a column to the data frame to specify if they are UP- or DOWN- regulated (log2fc respectively positive or negative)
dge_res_df$diffexpressed <- "NO"

# Clasify gene using treshold
dge_res_df$diffexpressed[dge_res_df$log2FoldChange > logfc & dge_res_df$pval < pval] <- "UP"
dge_res_df$diffexpressed[dge_res_df$log2FoldChange < -logfc & dge_res_df$pval < pval] <- "DOWN"

#Filter down genes
overexpressed_genes <- subset(dge_res_df, diffexpressed!="NO")  

dge_res_df
```

```{r}
plotDispEsts(dds)
```

## Volcano Plot

```{r}
#| label: VOlcano plot
ggplot(data = dge_res_df, aes(x = log2FoldChange, y = -log10(padj), col = diffexpressed)) +
  geom_vline(xintercept = c(-logfc,logfc), col = "gray", linetype = 'dashed') +
  geom_hline(yintercept = -log10(pval), col = "gray", linetype = 'dashed') + 
  geom_point(size = 2) + 
  scale_color_manual(values = c("#00AFBB", "grey", "#bb0c00"), 
                     labels = c("Downregulated", "Not significant", "Upregulated")) + 
  coord_cartesian(ylim = c(0, 200), xlim = c(-10, 10)) + 
  labs(color = 'Genes', #legend_title, 
       x = expression("log"[2]*"FC"), 
       y = expression("-log"[10]*"p-adj")) + 
  scale_x_continuous(breaks = seq(-10, 10, 2)) +
  ggtitle('DGE bet') 
 
```


# PCA

```{r}
# transpose a table
to_PCA <- norm_counts %>% t()  

#Run PCA
pca <- prcomp(to_PCA) #run PCA (the result is a list)

#Transform to dataframe
#Tu się coś wykrzacza ale wystarzy pujśc dalej i działa xd
pca_plot <- as.data.frame(pca$x) 
pca_plot
```

```{r}
#| label: PCA_1
# Combine PCA coordinates with metadata
pca_metadata_df <- cbind(pca_plot, metadata_filt)

# pca gender
p1 <- ggplot(pca_metadata_df, aes(x = PC1, y = PC2, color = gender)) +
  geom_point(size = 3, alpha = 0.8) +
  labs(title = "Gender", color = "Gender") +
  theme_minimal(base_size = 12)

#pca affect
p2 <- ggplot(pca_metadata_df, aes(x = PC1, y = PC2, color = affected)) +
  geom_point(size = 3, alpha = 0.8) +
  labs(title = "Affected", color = "Affected") +
  theme_minimal(base_size = 12)

#pca bmi)
p3 <- ggplot(pca_metadata_df, aes(x = PC1, y = PC2, color = bmi)) +
  geom_point(size = 3, alpha = 0.8) +
  scale_color_gradient(low = "lightblue", high = "darkblue") +
  labs(title = "BMI", color = "BMI") +
  theme_minimal(base_size = 12)

# pca age
p4 <- ggplot(pca_metadata_df, aes(x = PC1, y = PC2, color = age)) +
  geom_point(size = 3, alpha = 0.8) +
  scale_color_gradient(low = "lightgreen", high = "darkgreen") +
  labs(title = "Age", color = "Age") +
  theme_minimal(base_size = 12)

# Combine 4 plots into 2x2 grid
(p1 | p2) / (p3 | p4)
```

![]()

# Gene function

```{r}
library(msigdbr)
library(clusterProfiler)
```

```{r}
dge_res_df <- read.csv("data/dge_res_df.csv", row.names = 1)
```

```{r}
name_of_gene_set <- 'C7'
gene_sets_df <- msigdbr(species = "Homo sapiens", category = name_of_gene_set)
head(gene_sets_df)
```

```{r}
gene_sets <- gene_sets_df %>%
  dplyr::select(gs_name, ensembl_gene)
```

```{r}
df_sig <- dge_res_df[which(dge_res_df$diffexpressed!="NO"),]
deg_results_list <- split(df_sig, df_sig$diffexpressed)
```

```{r}
background_genes <- data_normalized %>% dplyr::select(Geneid)
```

```{r}
## Run ClusterProfiler
# Settings

name_of_comparison <- 'severevshealthy'
padj_cutoff <- 0.05
genecount_cutoff <- 5
filename_prefix <- paste0(name_of_comparison, '_', name_of_gene_set)
```

```{r}
# Run clusterProfiler on each sub-dataframe
res <- lapply(names(deg_results_list),
              function(x) enricher(gene = rownames(deg_results_list[[x]]),
                                   TERM2GENE = gene_sets,
                                   universe = background_genes,
                                   qvalueCutoff = 0.2))
names(res) <- names(deg_results_list)

cat("Enrichment analysis results:\n")
for(i in names(res)) {
  cat(paste(i, ":", nrow(as.data.frame(res[[i]])), "enriched pathways\n"))
}
```

```{r}
#Convert the enrichResults to a dataframe with the pathways
res_df <- lapply(names(res), function(x) rbind(res[[x]]@result))
names(res_df) <- names(res)
res_df <- do.call(rbind, res_df)
head(res_df)
```

```{r}
write.csv(res_df, "results.csv")
```

Differential expression analysis revealed that the upregulated gene set was dominated by non-coding transcripts, including long non-coding RNAs (lncRNAs), Y RNAs, and several transcribed pseudogenes (e.g. PRKY, ANOS2P, TXLNGY). Most of these transcripts are poorly characterized and lack evidence for protein-coding potential or involvement in known biological pathways. In contrast, the downregulated genes included protein-coding genes associated with cytoskeletal organization (MAP7D2), regulation of protein phosphatase activity (PPP1R2C), and cell cycle–related processes such as mitotic spindle control and chromatid segregation (ERCC6L). Additionally, two key long non-coding RNAs involved in X-chromosome inactivation, TSIX and XIST, were significantly downregulated.

# Multiomics clustering SUMO

```{r}
imune_marker_raw=read.csv("data/immune_marker_data.tsv", sep = "\t")
imune_marker_raw
```

```{r}
#Filter out not use sample
imune_marker<- imune_marker_raw %>% filter(analysis_type=="initial") %>% slice (1:262)

#Change rownames
rownames(imune_marker) <- imune_marker$ID

#Save to file
to_save <- imune_marker %>% dplyr::select(!c(age, sex, BMI, status, analysis_type, batch_variable))
write_tsv(to_save, file = "sumo/immune_markers.tsv")

imune_marker_numeric <- imune_marker %>% dplyr::select(!c(ID, age, sex, BMI, status, analysis_type, batch_variable))
imune_marker_numeric
```

## Normalize data

```{r}
#Prepare data for sumo
#Handle missing values, fill NA with mean()
df_imput <- (
  apply(imune_marker_numeric, 1, function(x) {
    x[is.na(x)] <- mean(x, na.rm = TRUE)
    x
  })
)

#Feature-wise standardization (mean=0, sd=1)
df_scaled <- t(scale(t(df_imput)))

#Verify scaling
feature_means <- rowMeans(df_scaled)
feature_sds <- apply(df_scaled, 1, sd)

summary(feature_means)
summary(feature_sds)

# Write output
write.table(
  df_scaled,
  file = "sumo/immune_markers.tsv",
  sep = "\t",
  quote = FALSE,
  col.names = NA
)
t
```

### DGE data

```{r}
vsd <- vst(dds, blind = TRUE)
expr <- assay(vsd)

# Variable genes (recommended)
vars <- apply(expr, 1, var)
genes_use <- names(sort(vars, decreasing = TRUE))[1:2000]

expr <- expr[genes_use, ]
expr_scaled <- t(scale(t(expr)))

write.table(
  expr_scaled,
  "sumo/sumo_rnaseq.tsv",
  sep = "\t",
  quote = FALSE,
  col.names = NA
)

```

### Metadata

```{r}
#Create labels.tsv We need becouse of sumo evaluate
label_affected <- metadata_filt %>%
  rownames_to_column(var = "sample") %>%
  mutate(label = case_when(
    affected == "control"  ~ 0,
    affected == "affected" ~ 1,
    TRUE                   ~ NA_real_
  )) %>%
  dplyr::select(sample, label) %>%
  as_tibble()

label_gender <- metadata_filt %>%
  rownames_to_column(var = "sample") %>%
  mutate(label = case_when(
    gender == "male"  ~ 0,
    gender == "female" ~ 1,
    TRUE                   ~ NA_real_
  )) %>%
  dplyr::select(sample, label) %>%
  as_tibble()

label_affected
label_gender

write_tsv(label_affected, "sumo/labels_affected.tsv")
write_tsv(label_gender, "sumo/labels_gender.tsv")
```

## Run SUMO

```{bash}
#source /home/janek/miniconda3/bin/activate #required by Quarto to recognize conda's commands
#conda activate SUMO
#cd sumo
#sumo prepare -atol 0.1 -plot LAML.png immune_markers.tsv,sumo_rnaseq.tsv output.npz 
#cd ..
#conda deactivate #deactivate the environment
```

```{bash}
#source /home/janek/miniconda3/bin/activate #required by Quarto to recognize conda's commands
#conda activate SUMO
#cd sumo

#sumo run output.npz 2,4 LAML

#cd ..
#conda deactivate #deactivate the environment
```

```{bash}
#source /home/janek/miniconda3/bin/activate #required by Quarto to recognize conda's commands
#conda activate SUMO
#cd sumo

#sumo run output.npz 2,7 LAML -rep 10 -r 4 

#cd ..
#conda deactivate #deactivate the environment
```

```{bash}
#source /home/janek/miniconda3/bin/activate #required by Quarto to recognize conda's commands
#conda activate SUMO
#cd sumo
#sumo evaluate LAML/k2/clusters.tsv labels_affected.tsv
#cd ..
conda deactivate #deactivate the environment
```

How to inerprate? NMI (Normalized Mutual Information):

## Anylise SUMO output

```{r}
# Cluster assignments
clusters <- read.table("sumo/LAML/k2/clusters.tsv", header=TRUE)
clusters
# columns: Sample, Cluster
rownames(clusters)<-clusters$sample
# Merge clusters with metadata
df <- merge(clusters, metadata_filt, by='row.names')
head(df)
```

```{r}
table(df$label, df$affected)

```

```{r}
prop.table(table(df$label, df$affected), 1)

```

```{r}
fisher.test(table(df$label, df$affected))

```

```{r}
ggplot(df, aes(x=factor(label), fill=factor(affected))) +
  geom_bar(position="fill") +
  labs(x="Cluster", y="Proportion", fill="Affected") +
  scale_y_continuous(labels=scales::percent_format()) +
  theme_minimal()

```

# Save data for raport
```{r}
saveRDS(dds, "RData/dds.rds")
saveRDS(metadata_filt, "RData/metadata_filt.rds")
saveRDS(norm_counts, "RData/norm_counts.rds")
saveRDS(pca_df_before, "RData/pca_df_before.rds")
saveRDS(pca_plot, "RData/pca_plot.rds")

```



