<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Maya Śliwa">
<meta name="author" content="Igor Mokrytskyi">
<meta name="author" content="Jan Rysiak">

<title>HTS DGE depression</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="HTS_DGE_depression_files/libs/clipboard/clipboard.min.js"></script>
<script src="HTS_DGE_depression_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="HTS_DGE_depression_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="HTS_DGE_depression_files/libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="HTS_DGE_depression_files/libs/quarto-html/popper.min.js"></script>
<script src="HTS_DGE_depression_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="HTS_DGE_depression_files/libs/quarto-html/anchor.min.js"></script>
<link href="HTS_DGE_depression_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="HTS_DGE_depression_files/libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="HTS_DGE_depression_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="HTS_DGE_depression_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="HTS_DGE_depression_files/libs/bootstrap/bootstrap-5861a310146c1438992d080840ff131b.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">


</head>

<body class="quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-full toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="6">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#download-filter-and-prepare-data" id="toc-download-filter-and-prepare-data" class="nav-link active" data-scroll-target="#download-filter-and-prepare-data">Download, filter and prepare data</a>
  <ul>
  <li><a href="#download-data" id="toc-download-data" class="nav-link" data-scroll-target="#download-data">Download data</a></li>
  <li><a href="#prepare-metadata" id="toc-prepare-metadata" class="nav-link" data-scroll-target="#prepare-metadata">Prepare metadata</a></li>
  <li><a href="#read-and-filter-count-matrix" id="toc-read-and-filter-count-matrix" class="nav-link" data-scroll-target="#read-and-filter-count-matrix">Read and filter count matrix</a></li>
  <li><a href="#read-and-filter-normalized-data" id="toc-read-and-filter-normalized-data" class="nav-link" data-scroll-target="#read-and-filter-normalized-data">Read and filter Normalized data</a></li>
  <li><a href="#inspect-data" id="toc-inspect-data" class="nav-link" data-scroll-target="#inspect-data">Inspect data</a>
  <ul>
  <li><a href="#checkpoint" id="toc-checkpoint" class="nav-link" data-scroll-target="#checkpoint">Checkpoint</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#dge-deseq2" id="toc-dge-deseq2" class="nav-link" data-scroll-target="#dge-deseq2">DGE DESeq2</a>
  <ul>
  <li><a href="#volcano-plot" id="toc-volcano-plot" class="nav-link" data-scroll-target="#volcano-plot">Volcano Plot</a></li>
  </ul></li>
  <li><a href="#pca" id="toc-pca" class="nav-link" data-scroll-target="#pca">PCA</a></li>
  <li><a href="#gene-function" id="toc-gene-function" class="nav-link" data-scroll-target="#gene-function">Gene function</a></li>
  <li><a href="#multiomics-clustering-sumo" id="toc-multiomics-clustering-sumo" class="nav-link" data-scroll-target="#multiomics-clustering-sumo">Multiomics clustering SUMO</a>
  <ul>
  <li><a href="#normalize-data" id="toc-normalize-data" class="nav-link" data-scroll-target="#normalize-data">Normalize data</a>
  <ul>
  <li><a href="#dge-data" id="toc-dge-data" class="nav-link" data-scroll-target="#dge-data">DGE data</a></li>
  <li><a href="#metadata" id="toc-metadata" class="nav-link" data-scroll-target="#metadata">Metadata</a></li>
  </ul></li>
  <li><a href="#run-sumo" id="toc-run-sumo" class="nav-link" data-scroll-target="#run-sumo">Run SUMO</a></li>
  <li><a href="#anylise-sumo-output" id="toc-anylise-sumo-output" class="nav-link" data-scroll-target="#anylise-sumo-output">Anylise SUMO output</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content column-page-right" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">HTS DGE depression</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>


<div class="quarto-title-meta-author column-page-right">
  <div class="quarto-title-meta-heading">Authors</div>
  <div class="quarto-title-meta-heading">Affiliation</div>
  
    <div class="quarto-title-meta-contents">
    <p class="author">Maya Śliwa </p>
  </div>
  <div class="quarto-title-meta-contents">
      </div>
    <div class="quarto-title-meta-contents">
    <p class="author">Igor Mokrytskyi </p>
  </div>
  <div class="quarto-title-meta-contents">
      </div>
    <div class="quarto-title-meta-contents">
    <p class="author">Jan Rysiak </p>
  </div>
  <div class="quarto-title-meta-contents">
      </div>
  </div>

<div class="quarto-title-meta column-page-right">

      
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">3 Oct yyyy</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Load Packages</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GGally)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GEOquery)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(apeglm)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2) </span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)  <span class="co"># for combining plots</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationDbi)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<section id="download-filter-and-prepare-data" class="level1">
<h1>Download, filter and prepare data</h1>
<section id="download-data" class="level2">
<h2 class="anchored" data-anchor-id="download-data">Download data</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Metadate</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co">#wget --directory-prefix=data https://ftp.ncbi.nlm.nih.gov/geo/series/GSE260nnn/GSE260603/matrix/GSE260603_series_matrix.txt.gz</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Rawcounts for analyzis</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co">#wget --directory-prefix=data https://ftp.ncbi.nlm.nih.gov/geo/series/GSE260nnn/GSE260603/suppl/GSE260603%5Fraw%5Fcounts.tsv.gz</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Normalize data</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co">#wget --directory-prefix=data https://ftp.ncbi.nlm.nih.gov/geo/series/GSE260nnn/GSE260603/suppl/GSE260603%5Fnormalized%5Fdata.tsv.gz</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="prepare-metadata" class="level2">
<h2 class="anchored" data-anchor-id="prepare-metadata">Prepare metadata</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Read metadata and inspect</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>gse <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="at">filename=</span><span class="st">"data/GSE260603_series_matrix.txt.gz"</span>, <span class="at">GSEMatrix=</span><span class="cn">FALSE</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">pData</span>(gse)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(metadata)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Rename columns and extract nedeed columns</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Set new name</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>old_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"age:ch1"</span>, <span class="st">"Sex:ch1"</span>, <span class="st">"bmi:ch1"</span>, <span class="st">"characteristics_ch1.4"</span>)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>new_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"age"</span>, <span class="st">"gender"</span>, <span class="st">"bmi"</span>, <span class="st">"affected"</span>)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Rename</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(metadata)[<span class="fu">match</span>(old_names, <span class="fu">colnames</span>(metadata))] <span class="ot">&lt;-</span> new_names</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Deal with brake row. For 'GSM8120701' there is some ERROR in data. </span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (metadata[<span class="st">'GSM8120701'</span>, <span class="dv">13</span>] <span class="sc">==</span>   <span class="st">"status: affected"</span>){</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  metadata[<span class="st">'GSM8120701'</span>, <span class="dv">14</span><span class="sc">:</span><span class="dv">20</span>] <span class="ot">&lt;-</span> metadata[<span class="st">'GSM8120701'</span>, <span class="dv">13</span><span class="sc">:</span><span class="dv">19</span>] </span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  metadata[<span class="st">'GSM8120701'</span>, <span class="dv">13</span>]<span class="ot">&lt;-</span><span class="cn">NaN</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(metadata[<span class="fu">c</span>(<span class="st">'GSM8120701'</span>, <span class="st">'GSM8120825'</span>, <span class="st">'GSM8120638'</span>), <span class="dv">13</span><span class="sc">:</span><span class="dv">20</span>])</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Filter, check and "civilize"  metadata</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Chose columns</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>metadata_filt<span class="ot">&lt;-</span>metadata[<span class="fu">c</span>(<span class="st">'title'</span>, <span class="st">'age'</span>, <span class="st">'gender'</span>, <span class="st">'bmi'</span>,<span class="st">'affected'</span>)]</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co">#New rownems</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>metadata_filt<span class="sc">$</span>title <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*(sample_[0-9]+).*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, metadata_filt<span class="sc">$</span>title)</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Drop rows with "status: affected; for replication analysis"</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>metadata_filt<span class="ot">&lt;-</span><span class="fu">subset</span>(metadata_filt, affected<span class="sc">!=</span><span class="st">"status: affected; for replication analysis"</span>)</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Chcnage afected</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>metadata_filt<span class="sc">$</span>affected <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"status: (.*)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, metadata_filt<span class="sc">$</span>affected)</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="co">#Set new rownames</span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(metadata_filt) <span class="ot">&lt;-</span> metadata_filt<span class="sc">$</span>title</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>metadata_filt<span class="ot">&lt;-</span><span class="fu">subset</span>(metadata_filt, <span class="at">select=</span><span class="sc">-</span>(title))</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(metadata_filt<span class="sc">$</span>bmi)</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert BMI and age to numeric</span></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>metadata_filt<span class="sc">$</span>bmi <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(metadata_filt<span class="sc">$</span>bmi)</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>metadata_filt<span class="sc">$</span>age <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(metadata_filt<span class="sc">$</span>age))</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>metadata_filt</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Visual check of metadata</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>bmi <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(metadata_filt, <span class="fu">aes</span>(<span class="at">x=</span>bmi)) <span class="sc">+</span> </span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>()</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>age <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(metadata_filt, <span class="fu">aes</span>(<span class="at">x=</span>age)) <span class="sc">+</span> </span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>()</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>(bmi<span class="sc">|</span>age)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Count male and female</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>summary_table <span class="ot">&lt;-</span> metadata_filt <span class="sc">%&gt;%</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(affected, gender) <span class="sc">%&gt;%</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>summary_table</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot </span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(summary_table, <span class="fu">aes</span>(<span class="at">x =</span> affected, <span class="at">y =</span> n, <span class="at">fill =</span> gender)) <span class="sc">+</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Count of male and female in groups"</span>,</span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Group"</span>,</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Liczba próbek"</span>,</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Gender"</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="read-and-filter-count-matrix" class="level2">
<h2 class="anchored" data-anchor-id="read-and-filter-count-matrix">Read and filter count matrix</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Get count matrix</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">"data/GSE260603_raw_counts.tsv.gz"</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Convert to array </span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>gene_ids <span class="ot">&lt;-</span> counts[[<span class="dv">1</span>]]</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>count_matrix_raw <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(counts[, <span class="sc">-</span><span class="dv">1</span>])</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(count_matrix_raw) <span class="ot">&lt;-</span> gene_ids</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(count_matrix_raw)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Filter out gene on chromosome Y</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Map Gene to chromosome</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>chr_map <span class="ot">&lt;-</span> AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  org.Hs.eg.db,</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">keys =</span> <span class="fu">rownames</span>(count_matrix_raw),</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="st">"CHR"</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out Y</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>genes_Y <span class="ot">&lt;-</span> chr_map<span class="sc">$</span>ENSEMBL[chr_map<span class="sc">$</span>CHR <span class="sc">==</span> <span class="st">"Y"</span>]</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>genes_Y<span class="ot">&lt;-</span><span class="fu">unique</span>(genes_Y)</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out genes form count matrix</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>count_matrix <span class="ot">&lt;-</span> count_matrix_raw[</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>  <span class="sc">!</span><span class="fu">rownames</span>(count_matrix_raw) <span class="sc">%in%</span> genes_Y,</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>]</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="read-and-filter-normalized-data" class="level2">
<h2 class="anchored" data-anchor-id="read-and-filter-normalized-data">Read and filter Normalized data</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>data_normalized <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="at">file =</span> <span class="st">"data/GSE260603_normalized_data.tsv"</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>data_normalized </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Transpose</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>df_t <span class="ot">&lt;-</span> <span class="fu">t</span>(data_normalized)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>df_t <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(df_t)</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df_t) <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unlist</span>(df_t[<span class="dv">1</span>,]))</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>df_t<span class="ot">&lt;-</span>df_t[<span class="sc">-</span><span class="dv">1</span>,]</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>df_t</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>expr <span class="ot">&lt;-</span> data_normalized <span class="sc">%&gt;%</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(expr) <span class="ot">&lt;-</span> expr<span class="sc">$</span>Geneid</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>expr<span class="sc">$</span>Geneid <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>expr <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(expr)</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="fu">storage.mode</span>(expr) <span class="ot">&lt;-</span> <span class="st">"numeric"</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(expr), <span class="at">center =</span> <span class="cn">TRUE</span>, <span class="at">scale. =</span> <span class="cn">TRUE</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># data frame</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>pca_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pca<span class="sc">$</span>x) <span class="sc">%&gt;%</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">rownames</span>(.))</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>pca_df</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="inspect-data" class="level2">
<h2 class="anchored" data-anchor-id="inspect-data">Inspect data</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine PCA coordinates with metadata</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>pca_metadata_df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(pca_df, metadata_filt)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="co"># pca gender</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_metadata_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> gender)) <span class="sc">+</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Gender"</span>, <span class="at">color =</span> <span class="st">"Gender"</span>) <span class="sc">+</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>)</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="co">#pca affect</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_metadata_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> affected)) <span class="sc">+</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Affected"</span>, <span class="at">color =</span> <span class="st">"Affected"</span>) <span class="sc">+</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>)</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a><span class="co">#pca bmi)</span></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_metadata_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> bmi)) <span class="sc">+</span></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">"lightblue"</span>, <span class="at">high =</span> <span class="st">"darkblue"</span>) <span class="sc">+</span></span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"BMI"</span>, <span class="at">color =</span> <span class="st">"BMI"</span>) <span class="sc">+</span></span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>)</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a><span class="co"># pca age</span></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_metadata_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> age)) <span class="sc">+</span></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">"lightgreen"</span>, <span class="at">high =</span> <span class="st">"darkgreen"</span>) <span class="sc">+</span></span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Age"</span>, <span class="at">color =</span> <span class="st">"Age"</span>) <span class="sc">+</span></span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>)</span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine 4 plots into 2x2 grid</span></span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a>(p1 <span class="sc">|</span> p2) <span class="sc">/</span> (p3 <span class="sc">|</span> p4)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<section id="checkpoint" class="level3">
<h3 class="anchored" data-anchor-id="checkpoint">Checkpoint</h3>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Save raw data</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="co">#save(metadata, counts, file = "RData/input_raw.RData")</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co">#SAVE DATA load for dds </span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="co">#save(metadata_filt, count_matrix, file = "RData/input_for_DGE_all.RData")</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="co">#load("RData/input_for_DGE_all.RData")</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
</section>
</section>
<section id="dge-deseq2" class="level1">
<h1>DGE DESeq2</h1>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Perform DGE</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">countData =</span> count_matrix,</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">colData =</span> metadata_filt,</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">design =</span> <span class="sc">~</span> affected <span class="sc">+</span> gender</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[<span class="fu">rowSums</span>(<span class="fu">counts</span>(dds)) <span class="sc">&gt;</span> <span class="dv">10</span>, ]</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Checkpoint</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="co">#save(dds, file="RData/dds_all.Rdata")</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p><img src=".png" class="img-fluid">load(“RData/dds_all.Rdata”)</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Set pval and logfc</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>pval<span class="ot">=</span><span class="fl">0.01</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>logfc<span class="ot">=</span><span class="dv">1</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Normalize data</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>norm_counts <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">counts</span>(dds, <span class="at">normalized =</span> T)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Get result</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds)</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Convert to data frame</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>dge_res_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(res)</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Drop row with Na in padj</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>dge_res_df <span class="ot">&lt;-</span><span class="fu">drop_na</span>(dge_res_df)</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a column to the data frame to specify if they are UP- or DOWN- regulated (log2fc respectively positive or negative)</span></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>dge_res_df<span class="sc">$</span>diffexpressed <span class="ot">&lt;-</span> <span class="st">"NO"</span></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Clasify gene using treshold</span></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>dge_res_df<span class="sc">$</span>diffexpressed[dge_res_df<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> logfc <span class="sc">&amp;</span> dge_res_df<span class="sc">$</span>pval <span class="sc">&lt;</span> pval] <span class="ot">&lt;-</span> <span class="st">"UP"</span></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>dge_res_df<span class="sc">$</span>diffexpressed[dge_res_df<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span>logfc <span class="sc">&amp;</span> dge_res_df<span class="sc">$</span>pval <span class="sc">&lt;</span> pval] <span class="ot">&lt;-</span> <span class="st">"DOWN"</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a><span class="co">#Filter down genes</span></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>overexpressed_genes <span class="ot">&lt;-</span> <span class="fu">subset</span>(dge_res_df, diffexpressed<span class="sc">!=</span><span class="st">"NO"</span>)  </span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>dge_res_df</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDispEsts</span>(dds)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<section id="volcano-plot" class="level2">
<h2 class="anchored" data-anchor-id="volcano-plot">Volcano Plot</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> dge_res_df, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">col =</span> diffexpressed)) <span class="sc">+</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span>logfc,logfc), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span> </span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#00AFBB"</span>, <span class="st">"grey"</span>, <span class="st">"#bb0c00"</span>), </span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Downregulated"</span>, <span class="st">"Not significant"</span>, <span class="st">"Upregulated"</span>)) <span class="sc">+</span> </span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">200</span>), <span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="dv">10</span>)) <span class="sc">+</span> </span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">color =</span> <span class="st">'Genes'</span>, <span class="co">#legend_title, </span></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">"FC"</span>), </span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"-log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"p-adj"</span>)) <span class="sc">+</span> </span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="dv">10</span>, <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">'DGE bet'</span>) </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p><img src=".png" class="img-fluid"></p>
</section>
</section>
<section id="pca" class="level1">
<h1>PCA</h1>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># transpose a table</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>to_PCA <span class="ot">&lt;-</span> norm_counts <span class="sc">%&gt;%</span> <span class="fu">t</span>()  </span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Run PCA</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(to_PCA) <span class="co">#run PCA (the result is a list)</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Transform to dataframe</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Tu się coś wykrzacza ale wystarzy pujśc dalej i działa xd</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>pca_plot <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pca<span class="sc">$</span>x) </span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>pca_plot</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine PCA coordinates with metadata</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>pca_metadata_df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(pca_plot, metadata_filt)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a><span class="co"># pca gender</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_metadata_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> gender)) <span class="sc">+</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Gender"</span>, <span class="at">color =</span> <span class="st">"Gender"</span>) <span class="sc">+</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>)</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a><span class="co">#pca affect</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_metadata_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> affected)) <span class="sc">+</span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Affected"</span>, <span class="at">color =</span> <span class="st">"Affected"</span>) <span class="sc">+</span></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>)</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a><span class="co">#pca bmi)</span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_metadata_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> bmi)) <span class="sc">+</span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">"lightblue"</span>, <span class="at">high =</span> <span class="st">"darkblue"</span>) <span class="sc">+</span></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"BMI"</span>, <span class="at">color =</span> <span class="st">"BMI"</span>) <span class="sc">+</span></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>)</span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a><span class="co"># pca age</span></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_metadata_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> age)) <span class="sc">+</span></span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">"lightgreen"</span>, <span class="at">high =</span> <span class="st">"darkgreen"</span>) <span class="sc">+</span></span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Age"</span>, <span class="at">color =</span> <span class="st">"Age"</span>) <span class="sc">+</span></span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>)</span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine 4 plots into 2x2 grid</span></span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a>(p1 <span class="sc">|</span> p2) <span class="sc">/</span> (p3 <span class="sc">|</span> p4)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p><img src=".png" class="img-fluid"></p>
</section>
<section id="gene-function" class="level1">
<h1>Gene function</h1>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(msigdbr)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>dge_res_df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/dge_res_df.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>name_of_gene_set <span class="ot">&lt;-</span> <span class="st">'C7'</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>gene_sets_df <span class="ot">&lt;-</span> <span class="fu">msigdbr</span>(<span class="at">species =</span> <span class="st">"Homo sapiens"</span>, <span class="at">category =</span> name_of_gene_set)</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gene_sets_df)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>gene_sets <span class="ot">&lt;-</span> gene_sets_df <span class="sc">%&gt;%</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gs_name, ensembl_gene)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>df_sig <span class="ot">&lt;-</span> dge_res_df[<span class="fu">which</span>(dge_res_df<span class="sc">$</span>diffexpressed<span class="sc">!=</span><span class="st">"NO"</span>),]</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>deg_results_list <span class="ot">&lt;-</span> <span class="fu">split</span>(df_sig, df_sig<span class="sc">$</span>diffexpressed)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>background_genes <span class="ot">&lt;-</span> data_normalized <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Geneid)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Run ClusterProfiler</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Settings</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>name_of_comparison <span class="ot">&lt;-</span> <span class="st">'severevshealthy'</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>padj_cutoff <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>genecount_cutoff <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>filename_prefix <span class="ot">&lt;-</span> <span class="fu">paste0</span>(name_of_comparison, <span class="st">'_'</span>, name_of_gene_set)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run clusterProfiler on each sub-dataframe</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">names</span>(deg_results_list),</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>              <span class="cf">function</span>(x) <span class="fu">enricher</span>(<span class="at">gene =</span> <span class="fu">rownames</span>(deg_results_list[[x]]),</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">TERM2GENE =</span> gene_sets,</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">universe =</span> background_genes,</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">qvalueCutoff =</span> <span class="fl">0.2</span>))</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(res) <span class="ot">&lt;-</span> <span class="fu">names</span>(deg_results_list)</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Enrichment analysis results:</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">names</span>(res)) {</span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">paste</span>(i, <span class="st">":"</span>, <span class="fu">nrow</span>(<span class="fu">as.data.frame</span>(res[[i]])), <span class="st">"enriched pathways</span><span class="sc">\n</span><span class="st">"</span>))</span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Convert the enrichResults to a dataframe with the pathways</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>res_df <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">names</span>(res), <span class="cf">function</span>(x) <span class="fu">rbind</span>(res[[x]]<span class="sc">@</span>result))</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(res_df) <span class="ot">&lt;-</span> <span class="fu">names</span>(res)</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>res_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, res_df)</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_df)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(res_df, <span class="st">"results.csv"</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Differential expression analysis revealed that the upregulated gene set was dominated by non-coding transcripts, including long non-coding RNAs (lncRNAs), Y RNAs, and several transcribed pseudogenes (e.g.&nbsp;PRKY, ANOS2P, TXLNGY). Most of these transcripts are poorly characterized and lack evidence for protein-coding potential or involvement in known biological pathways. In contrast, the downregulated genes included protein-coding genes associated with cytoskeletal organization (MAP7D2), regulation of protein phosphatase activity (PPP1R2C), and cell cycle–related processes such as mitotic spindle control and chromatid segregation (ERCC6L). Additionally, two key long non-coding RNAs involved in X-chromosome inactivation, TSIX and XIST, were significantly downregulated.</p>
</section>
<section id="multiomics-clustering-sumo" class="level1">
<h1>Multiomics clustering SUMO</h1>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>imune_marker_raw<span class="ot">=</span><span class="fu">read.csv</span>(<span class="st">"data/immune_marker_data.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>imune_marker_raw</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Filter out not use sample</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>imune_marker<span class="ot">&lt;-</span> imune_marker_raw <span class="sc">%&gt;%</span> <span class="fu">filter</span>(analysis_type<span class="sc">==</span><span class="st">"initial"</span>) <span class="sc">%&gt;%</span> <span class="fu">slice</span> (<span class="dv">1</span><span class="sc">:</span><span class="dv">262</span>)</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Change rownames</span></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(imune_marker) <span class="ot">&lt;-</span> imune_marker<span class="sc">$</span>ID</span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Save to file</span></span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>to_save <span class="ot">&lt;-</span> imune_marker <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(age, sex, BMI, status, analysis_type, batch_variable))</span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(to_save, <span class="at">file =</span> <span class="st">"sumo/immune_markers.tsv"</span>)</span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a>imune_marker_numeric <span class="ot">&lt;-</span> imune_marker <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(ID, age, sex, BMI, status, analysis_type, batch_variable))</span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a>imune_marker_numeric</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<section id="normalize-data" class="level2">
<h2 class="anchored" data-anchor-id="normalize-data">Normalize data</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Prepare data for sumo</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Handle missing values, fill NA with mean()</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>df_imput <span class="ot">&lt;-</span> (</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">apply</span>(imune_marker_numeric, <span class="dv">1</span>, <span class="cf">function</span>(x) {</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>    x[<span class="fu">is.na</span>(x)] <span class="ot">&lt;-</span> <span class="fu">mean</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>    x</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Feature-wise standardization (mean=0, sd=1)</span></span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>df_scaled <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(df_imput)))</span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Verify scaling</span></span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a>feature_means <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(df_scaled)</span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a>feature_sds <span class="ot">&lt;-</span> <span class="fu">apply</span>(df_scaled, <span class="dv">1</span>, sd)</span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-17"><a href="#cb38-17" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(feature_means)</span>
<span id="cb38-18"><a href="#cb38-18" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(feature_sds)</span>
<span id="cb38-19"><a href="#cb38-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-20"><a href="#cb38-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Write output</span></span>
<span id="cb38-21"><a href="#cb38-21" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(</span>
<span id="cb38-22"><a href="#cb38-22" aria-hidden="true" tabindex="-1"></a>  df_scaled,</span>
<span id="cb38-23"><a href="#cb38-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"sumo/immune_markers.tsv"</span>,</span>
<span id="cb38-24"><a href="#cb38-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb38-25"><a href="#cb38-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">quote =</span> <span class="cn">FALSE</span>,</span>
<span id="cb38-26"><a href="#cb38-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">col.names =</span> <span class="cn">NA</span></span>
<span id="cb38-27"><a href="#cb38-27" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb38-28"><a href="#cb38-28" aria-hidden="true" tabindex="-1"></a>t</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<section id="dge-data" class="level3">
<h3 class="anchored" data-anchor-id="dge-data">DGE data</h3>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>expr <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd)</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Variable genes (recommended)</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>vars <span class="ot">&lt;-</span> <span class="fu">apply</span>(expr, <span class="dv">1</span>, var)</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>genes_use <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">sort</span>(vars, <span class="at">decreasing =</span> <span class="cn">TRUE</span>))[<span class="dv">1</span><span class="sc">:</span><span class="dv">2000</span>]</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>expr <span class="ot">&lt;-</span> expr[genes_use, ]</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>expr_scaled <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(expr)))</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(</span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a>  expr_scaled,</span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">"sumo/sumo_rnaseq.tsv"</span>,</span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">quote =</span> <span class="cn">FALSE</span>,</span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">col.names =</span> <span class="cn">NA</span></span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
<section id="metadata" class="level3">
<h3 class="anchored" data-anchor-id="metadata">Metadata</h3>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Create labels.tsv We need becouse of sumo evaluate</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>label_affected <span class="ot">&lt;-</span> metadata_filt <span class="sc">%&gt;%</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">case_when</span>(</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>    affected <span class="sc">==</span> <span class="st">"control"</span>  <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>    affected <span class="sc">==</span> <span class="st">"affected"</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span>                   <span class="sc">~</span> <span class="cn">NA_real_</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, label) <span class="sc">%&gt;%</span></span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>label_gender <span class="ot">&lt;-</span> metadata_filt <span class="sc">%&gt;%</span></span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">case_when</span>(</span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>    gender <span class="sc">==</span> <span class="st">"male"</span>  <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>    gender <span class="sc">==</span> <span class="st">"female"</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span>                   <span class="sc">~</span> <span class="cn">NA_real_</span></span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span></span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, label) <span class="sc">%&gt;%</span></span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a>label_affected</span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a>label_gender</span>
<span id="cb40-24"><a href="#cb40-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-25"><a href="#cb40-25" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(label_affected, <span class="st">"sumo/labels_affected.tsv"</span>)</span>
<span id="cb40-26"><a href="#cb40-26" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(label_gender, <span class="st">"sumo/labels_gender.tsv"</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
</section>
<section id="run-sumo" class="level2">
<h2 class="anchored" data-anchor-id="run-sumo">Run SUMO</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb41"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co">#source /home/janek/miniconda3/bin/activate #required by Quarto to recognize conda's commands</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="co">#conda activate SUMO</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="co">#cd sumo</span></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a><span class="co">#sumo prepare -atol 0.1 -plot LAML.png immune_markers.tsv,sumo_rnaseq.tsv output.npz </span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a><span class="co">#cd ..</span></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a><span class="co">#conda deactivate #deactivate the environment</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb42"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co">#source /home/janek/miniconda3/bin/activate #required by Quarto to recognize conda's commands</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="co">#conda activate SUMO</span></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a><span class="co">#cd sumo</span></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a><span class="co">#sumo run output.npz 2,4 LAML</span></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a><span class="co">#cd ..</span></span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a><span class="co">#conda deactivate #deactivate the environment</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb43"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="co">#source /home/janek/miniconda3/bin/activate #required by Quarto to recognize conda's commands</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="co">#conda activate SUMO</span></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a><span class="co">#cd sumo</span></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a><span class="co">#sumo run output.npz 2,7 LAML -rep 10 -r 4 </span></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a><span class="co">#cd ..</span></span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a><span class="co">#conda deactivate #deactivate the environment</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb44"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> /home/janek/miniconda3/bin/activate <span class="co">#required by Quarto to recognize conda's commands</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate SUMO</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> sumo</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a><span class="ex">sumo</span> evaluate LAML/k2/clusters.tsv labels_affected.tsv</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ..</span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate <span class="co">#deactivate the environment</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>How to inerprate? NMI (Normalized Mutual Information):</p>
</section>
<section id="anylise-sumo-output" class="level2">
<h2 class="anchored" data-anchor-id="anylise-sumo-output">Anylise SUMO output</h2>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Cluster assignments</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>clusters <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"sumo/LAML/k2/clusters.tsv"</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>clusters</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a><span class="co"># columns: Sample, Cluster</span></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(clusters)<span class="ot">&lt;-</span>clusters<span class="sc">$</span>sample</span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge clusters with metadata</span></span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">merge</span>(clusters, metadata_filt, <span class="at">by=</span><span class="st">'row.names'</span>)</span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(df)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(df<span class="sc">$</span>label, df<span class="sc">$</span>affected)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">prop.table</span>(<span class="fu">table</span>(df<span class="sc">$</span>label, df<span class="sc">$</span>affected), <span class="dv">1</span>)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fisher.test</span>(<span class="fu">table</span>(df<span class="sc">$</span>label, df<span class="sc">$</span>affected))</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">factor</span>(label), <span class="at">fill=</span><span class="fu">factor</span>(affected))) <span class="sc">+</span></span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">position=</span><span class="st">"fill"</span>) <span class="sc">+</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Cluster"</span>, <span class="at">y=</span><span class="st">"Proportion"</span>, <span class="at">fill=</span><span class="st">"Affected"</span>) <span class="sc">+</span></span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels=</span>scales<span class="sc">::</span><span class="fu">percent_format</span>()) <span class="sc">+</span></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<!-- -->

</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb50" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "HTS DGE depression"</span></span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Maya Śliwa</span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Igor Mokrytskyi</span></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Jan Rysiak</span></span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> 2026-10-07</span></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a><span class="an">date-format:</span><span class="co"> "d MMM yyyy"</span></span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a><span class="an">affiliations:</span></span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a><span class="co">  - id: JU</span></span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a><span class="co">    name: Jagiellonian University</span></span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span><span class="co"> </span></span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a><span class="co">    df-print: kable</span></span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a><span class="co">    tidy: false</span></span>
<span id="cb50-16"><a href="#cb50-16" aria-hidden="true" tabindex="-1"></a><span class="co">    embed-resources: false</span></span>
<span id="cb50-17"><a href="#cb50-17" aria-hidden="true" tabindex="-1"></a><span class="co">    theme: lumen</span></span>
<span id="cb50-18"><a href="#cb50-18" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true</span></span>
<span id="cb50-19"><a href="#cb50-19" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-location: left</span></span>
<span id="cb50-20"><a href="#cb50-20" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-depth: 6</span></span>
<span id="cb50-21"><a href="#cb50-21" aria-hidden="true" tabindex="-1"></a><span class="co">    toc-expand: 6</span></span>
<span id="cb50-22"><a href="#cb50-22" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools:</span></span>
<span id="cb50-23"><a href="#cb50-23" aria-hidden="true" tabindex="-1"></a><span class="co">      source: true</span></span>
<span id="cb50-24"><a href="#cb50-24" aria-hidden="true" tabindex="-1"></a><span class="co">      toggle: true</span></span>
<span id="cb50-25"><a href="#cb50-25" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: false</span></span>
<span id="cb50-26"><a href="#cb50-26" aria-hidden="true" tabindex="-1"></a><span class="co">    code-summary: "Show the code"</span></span>
<span id="cb50-27"><a href="#cb50-27" aria-hidden="true" tabindex="-1"></a><span class="co">    code-overflow: wrap</span></span>
<span id="cb50-28"><a href="#cb50-28" aria-hidden="true" tabindex="-1"></a><span class="co">    page-layout: full</span></span>
<span id="cb50-29"><a href="#cb50-29" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb50-30"><a href="#cb50-30" aria-hidden="true" tabindex="-1"></a><span class="co">  eval: false</span></span>
<span id="cb50-31"><a href="#cb50-31" aria-hidden="true" tabindex="-1"></a><span class="co">  cache: true</span></span>
<span id="cb50-32"><a href="#cb50-32" aria-hidden="true" tabindex="-1"></a><span class="an">editor:</span><span class="co"> visual</span></span>
<span id="cb50-33"><a href="#cb50-33" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb50-34"><a href="#cb50-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-37"><a href="#cb50-37" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-38"><a href="#cb50-38" aria-hidden="true" tabindex="-1"></a><span class="co">#Load Packages</span></span>
<span id="cb50-39"><a href="#cb50-39" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GGally)</span>
<span id="cb50-40"><a href="#cb50-40" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb50-41"><a href="#cb50-41" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb50-42"><a href="#cb50-42" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GEOquery)</span>
<span id="cb50-43"><a href="#cb50-43" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb50-44"><a href="#cb50-44" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb50-45"><a href="#cb50-45" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(apeglm)</span>
<span id="cb50-46"><a href="#cb50-46" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb50-47"><a href="#cb50-47" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2) </span>
<span id="cb50-48"><a href="#cb50-48" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)  <span class="co"># for combining plots</span></span>
<span id="cb50-49"><a href="#cb50-49" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb50-50"><a href="#cb50-50" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationDbi)</span>
<span id="cb50-51"><a href="#cb50-51" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-52"><a href="#cb50-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-53"><a href="#cb50-53" aria-hidden="true" tabindex="-1"></a><span class="fu"># Download, filter and prepare data</span></span>
<span id="cb50-54"><a href="#cb50-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-55"><a href="#cb50-55" aria-hidden="true" tabindex="-1"></a><span class="fu">## Download data</span></span>
<span id="cb50-56"><a href="#cb50-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-59"><a href="#cb50-59" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb50-60"><a href="#cb50-60" aria-hidden="true" tabindex="-1"></a><span class="co">#Metadate</span></span>
<span id="cb50-61"><a href="#cb50-61" aria-hidden="true" tabindex="-1"></a><span class="co">#wget --directory-prefix=data https://ftp.ncbi.nlm.nih.gov/geo/series/GSE260nnn/GSE260603/matrix/GSE260603_series_matrix.txt.gz</span></span>
<span id="cb50-62"><a href="#cb50-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-63"><a href="#cb50-63" aria-hidden="true" tabindex="-1"></a><span class="co">#Rawcounts for analyzis</span></span>
<span id="cb50-64"><a href="#cb50-64" aria-hidden="true" tabindex="-1"></a><span class="co">#wget --directory-prefix=data https://ftp.ncbi.nlm.nih.gov/geo/series/GSE260nnn/GSE260603/suppl/GSE260603%5Fraw%5Fcounts.tsv.gz</span></span>
<span id="cb50-65"><a href="#cb50-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-66"><a href="#cb50-66" aria-hidden="true" tabindex="-1"></a><span class="co">#Normalize data</span></span>
<span id="cb50-67"><a href="#cb50-67" aria-hidden="true" tabindex="-1"></a><span class="co">#wget --directory-prefix=data https://ftp.ncbi.nlm.nih.gov/geo/series/GSE260nnn/GSE260603/suppl/GSE260603%5Fnormalized%5Fdata.tsv.gz</span></span>
<span id="cb50-68"><a href="#cb50-68" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-69"><a href="#cb50-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-70"><a href="#cb50-70" aria-hidden="true" tabindex="-1"></a><span class="fu">## Prepare metadata</span></span>
<span id="cb50-71"><a href="#cb50-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-74"><a href="#cb50-74" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-75"><a href="#cb50-75" aria-hidden="true" tabindex="-1"></a><span class="co">#Read metadata and inspect</span></span>
<span id="cb50-76"><a href="#cb50-76" aria-hidden="true" tabindex="-1"></a>gse <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="at">filename=</span><span class="st">"data/GSE260603_series_matrix.txt.gz"</span>, <span class="at">GSEMatrix=</span><span class="cn">FALSE</span>)</span>
<span id="cb50-77"><a href="#cb50-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-78"><a href="#cb50-78" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">pData</span>(gse)</span>
<span id="cb50-79"><a href="#cb50-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-80"><a href="#cb50-80" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(metadata)</span>
<span id="cb50-81"><a href="#cb50-81" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-82"><a href="#cb50-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-85"><a href="#cb50-85" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-86"><a href="#cb50-86" aria-hidden="true" tabindex="-1"></a><span class="co">#Rename columns and extract nedeed columns</span></span>
<span id="cb50-87"><a href="#cb50-87" aria-hidden="true" tabindex="-1"></a><span class="co">#Set new name</span></span>
<span id="cb50-88"><a href="#cb50-88" aria-hidden="true" tabindex="-1"></a>old_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"age:ch1"</span>, <span class="st">"Sex:ch1"</span>, <span class="st">"bmi:ch1"</span>, <span class="st">"characteristics_ch1.4"</span>)</span>
<span id="cb50-89"><a href="#cb50-89" aria-hidden="true" tabindex="-1"></a>new_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"age"</span>, <span class="st">"gender"</span>, <span class="st">"bmi"</span>, <span class="st">"affected"</span>)</span>
<span id="cb50-90"><a href="#cb50-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-91"><a href="#cb50-91" aria-hidden="true" tabindex="-1"></a><span class="co">#Rename</span></span>
<span id="cb50-92"><a href="#cb50-92" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(metadata)[<span class="fu">match</span>(old_names, <span class="fu">colnames</span>(metadata))] <span class="ot">&lt;-</span> new_names</span>
<span id="cb50-93"><a href="#cb50-93" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-94"><a href="#cb50-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-97"><a href="#cb50-97" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-98"><a href="#cb50-98" aria-hidden="true" tabindex="-1"></a><span class="co">#Deal with brake row. For 'GSM8120701' there is some ERROR in data. </span></span>
<span id="cb50-99"><a href="#cb50-99" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (metadata[<span class="st">'GSM8120701'</span>, <span class="dv">13</span>] <span class="sc">==</span>   <span class="st">"status: affected"</span>){</span>
<span id="cb50-100"><a href="#cb50-100" aria-hidden="true" tabindex="-1"></a>  metadata[<span class="st">'GSM8120701'</span>, <span class="dv">14</span><span class="sc">:</span><span class="dv">20</span>] <span class="ot">&lt;-</span> metadata[<span class="st">'GSM8120701'</span>, <span class="dv">13</span><span class="sc">:</span><span class="dv">19</span>] </span>
<span id="cb50-101"><a href="#cb50-101" aria-hidden="true" tabindex="-1"></a>  metadata[<span class="st">'GSM8120701'</span>, <span class="dv">13</span>]<span class="ot">&lt;-</span><span class="cn">NaN</span></span>
<span id="cb50-102"><a href="#cb50-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(metadata[<span class="fu">c</span>(<span class="st">'GSM8120701'</span>, <span class="st">'GSM8120825'</span>, <span class="st">'GSM8120638'</span>), <span class="dv">13</span><span class="sc">:</span><span class="dv">20</span>])</span>
<span id="cb50-103"><a href="#cb50-103" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb50-104"><a href="#cb50-104" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-105"><a href="#cb50-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-108"><a href="#cb50-108" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-109"><a href="#cb50-109" aria-hidden="true" tabindex="-1"></a><span class="co">#Filter, check and "civilize"  metadata</span></span>
<span id="cb50-110"><a href="#cb50-110" aria-hidden="true" tabindex="-1"></a><span class="co">#Chose columns</span></span>
<span id="cb50-111"><a href="#cb50-111" aria-hidden="true" tabindex="-1"></a>metadata_filt<span class="ot">&lt;-</span>metadata[<span class="fu">c</span>(<span class="st">'title'</span>, <span class="st">'age'</span>, <span class="st">'gender'</span>, <span class="st">'bmi'</span>,<span class="st">'affected'</span>)]</span>
<span id="cb50-112"><a href="#cb50-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-113"><a href="#cb50-113" aria-hidden="true" tabindex="-1"></a><span class="co">#New rownems</span></span>
<span id="cb50-114"><a href="#cb50-114" aria-hidden="true" tabindex="-1"></a>metadata_filt<span class="sc">$</span>title <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">".*(sample_[0-9]+).*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, metadata_filt<span class="sc">$</span>title)</span>
<span id="cb50-115"><a href="#cb50-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-116"><a href="#cb50-116" aria-hidden="true" tabindex="-1"></a><span class="co">#Drop rows with "status: affected; for replication analysis"</span></span>
<span id="cb50-117"><a href="#cb50-117" aria-hidden="true" tabindex="-1"></a>metadata_filt<span class="ot">&lt;-</span><span class="fu">subset</span>(metadata_filt, affected<span class="sc">!=</span><span class="st">"status: affected; for replication analysis"</span>)</span>
<span id="cb50-118"><a href="#cb50-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-119"><a href="#cb50-119" aria-hidden="true" tabindex="-1"></a><span class="co">#Chcnage afected</span></span>
<span id="cb50-120"><a href="#cb50-120" aria-hidden="true" tabindex="-1"></a>metadata_filt<span class="sc">$</span>affected <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"status: (.*)"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, metadata_filt<span class="sc">$</span>affected)</span>
<span id="cb50-121"><a href="#cb50-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-122"><a href="#cb50-122" aria-hidden="true" tabindex="-1"></a><span class="co">#Set new rownames</span></span>
<span id="cb50-123"><a href="#cb50-123" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(metadata_filt) <span class="ot">&lt;-</span> metadata_filt<span class="sc">$</span>title</span>
<span id="cb50-124"><a href="#cb50-124" aria-hidden="true" tabindex="-1"></a>metadata_filt<span class="ot">&lt;-</span><span class="fu">subset</span>(metadata_filt, <span class="at">select=</span><span class="sc">-</span>(title))</span>
<span id="cb50-125"><a href="#cb50-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-126"><a href="#cb50-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-127"><a href="#cb50-127" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(metadata_filt<span class="sc">$</span>bmi)</span>
<span id="cb50-128"><a href="#cb50-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-129"><a href="#cb50-129" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert BMI and age to numeric</span></span>
<span id="cb50-130"><a href="#cb50-130" aria-hidden="true" tabindex="-1"></a>metadata_filt<span class="sc">$</span>bmi <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(metadata_filt<span class="sc">$</span>bmi)</span>
<span id="cb50-131"><a href="#cb50-131" aria-hidden="true" tabindex="-1"></a>metadata_filt<span class="sc">$</span>age <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(metadata_filt<span class="sc">$</span>age))</span>
<span id="cb50-132"><a href="#cb50-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-133"><a href="#cb50-133" aria-hidden="true" tabindex="-1"></a>metadata_filt</span>
<span id="cb50-134"><a href="#cb50-134" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-135"><a href="#cb50-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-138"><a href="#cb50-138" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-139"><a href="#cb50-139" aria-hidden="true" tabindex="-1"></a><span class="co">#Visual check of metadata</span></span>
<span id="cb50-140"><a href="#cb50-140" aria-hidden="true" tabindex="-1"></a>bmi <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(metadata_filt, <span class="fu">aes</span>(<span class="at">x=</span>bmi)) <span class="sc">+</span> </span>
<span id="cb50-141"><a href="#cb50-141" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>()</span>
<span id="cb50-142"><a href="#cb50-142" aria-hidden="true" tabindex="-1"></a>age <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(metadata_filt, <span class="fu">aes</span>(<span class="at">x=</span>age)) <span class="sc">+</span> </span>
<span id="cb50-143"><a href="#cb50-143" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>()</span>
<span id="cb50-144"><a href="#cb50-144" aria-hidden="true" tabindex="-1"></a>(bmi<span class="sc">|</span>age)</span>
<span id="cb50-145"><a href="#cb50-145" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-146"><a href="#cb50-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-149"><a href="#cb50-149" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-150"><a href="#cb50-150" aria-hidden="true" tabindex="-1"></a><span class="co">#Count male and female</span></span>
<span id="cb50-151"><a href="#cb50-151" aria-hidden="true" tabindex="-1"></a>summary_table <span class="ot">&lt;-</span> metadata_filt <span class="sc">%&gt;%</span></span>
<span id="cb50-152"><a href="#cb50-152" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(affected, gender) <span class="sc">%&gt;%</span></span>
<span id="cb50-153"><a href="#cb50-153" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb50-154"><a href="#cb50-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-155"><a href="#cb50-155" aria-hidden="true" tabindex="-1"></a>summary_table</span>
<span id="cb50-156"><a href="#cb50-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-157"><a href="#cb50-157" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot </span></span>
<span id="cb50-158"><a href="#cb50-158" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(summary_table, <span class="fu">aes</span>(<span class="at">x =</span> affected, <span class="at">y =</span> n, <span class="at">fill =</span> gender)) <span class="sc">+</span></span>
<span id="cb50-159"><a href="#cb50-159" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb50-160"><a href="#cb50-160" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb50-161"><a href="#cb50-161" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb50-162"><a href="#cb50-162" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Count of male and female in groups"</span>,</span>
<span id="cb50-163"><a href="#cb50-163" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Group"</span>,</span>
<span id="cb50-164"><a href="#cb50-164" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Liczba próbek"</span>,</span>
<span id="cb50-165"><a href="#cb50-165" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"Gender"</span></span>
<span id="cb50-166"><a href="#cb50-166" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb50-167"><a href="#cb50-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-168"><a href="#cb50-168" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-169"><a href="#cb50-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-170"><a href="#cb50-170" aria-hidden="true" tabindex="-1"></a><span class="fu">## Read and filter count matrix</span></span>
<span id="cb50-171"><a href="#cb50-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-174"><a href="#cb50-174" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-175"><a href="#cb50-175" aria-hidden="true" tabindex="-1"></a><span class="co">#Get count matrix</span></span>
<span id="cb50-176"><a href="#cb50-176" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">"data/GSE260603_raw_counts.tsv.gz"</span>)</span>
<span id="cb50-177"><a href="#cb50-177" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-178"><a href="#cb50-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-181"><a href="#cb50-181" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-182"><a href="#cb50-182" aria-hidden="true" tabindex="-1"></a><span class="co">#Convert to array </span></span>
<span id="cb50-183"><a href="#cb50-183" aria-hidden="true" tabindex="-1"></a>gene_ids <span class="ot">&lt;-</span> counts[[<span class="dv">1</span>]]</span>
<span id="cb50-184"><a href="#cb50-184" aria-hidden="true" tabindex="-1"></a>count_matrix_raw <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(counts[, <span class="sc">-</span><span class="dv">1</span>])</span>
<span id="cb50-185"><a href="#cb50-185" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(count_matrix_raw) <span class="ot">&lt;-</span> gene_ids</span>
<span id="cb50-186"><a href="#cb50-186" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(count_matrix_raw)</span>
<span id="cb50-187"><a href="#cb50-187" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-188"><a href="#cb50-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-191"><a href="#cb50-191" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-192"><a href="#cb50-192" aria-hidden="true" tabindex="-1"></a><span class="co">#Filter out gene on chromosome Y</span></span>
<span id="cb50-193"><a href="#cb50-193" aria-hidden="true" tabindex="-1"></a><span class="co">#Map Gene to chromosome</span></span>
<span id="cb50-194"><a href="#cb50-194" aria-hidden="true" tabindex="-1"></a>chr_map <span class="ot">&lt;-</span> AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(</span>
<span id="cb50-195"><a href="#cb50-195" aria-hidden="true" tabindex="-1"></a>  org.Hs.eg.db,</span>
<span id="cb50-196"><a href="#cb50-196" aria-hidden="true" tabindex="-1"></a>  <span class="at">keys =</span> <span class="fu">rownames</span>(count_matrix_raw),</span>
<span id="cb50-197"><a href="#cb50-197" aria-hidden="true" tabindex="-1"></a>  <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>,</span>
<span id="cb50-198"><a href="#cb50-198" aria-hidden="true" tabindex="-1"></a>  <span class="at">columns =</span> <span class="st">"CHR"</span></span>
<span id="cb50-199"><a href="#cb50-199" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb50-200"><a href="#cb50-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-201"><a href="#cb50-201" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out Y</span></span>
<span id="cb50-202"><a href="#cb50-202" aria-hidden="true" tabindex="-1"></a>genes_Y <span class="ot">&lt;-</span> chr_map<span class="sc">$</span>ENSEMBL[chr_map<span class="sc">$</span>CHR <span class="sc">==</span> <span class="st">"Y"</span>]</span>
<span id="cb50-203"><a href="#cb50-203" aria-hidden="true" tabindex="-1"></a>genes_Y<span class="ot">&lt;-</span><span class="fu">unique</span>(genes_Y)</span>
<span id="cb50-204"><a href="#cb50-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-205"><a href="#cb50-205" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out genes form count matrix</span></span>
<span id="cb50-206"><a href="#cb50-206" aria-hidden="true" tabindex="-1"></a>count_matrix <span class="ot">&lt;-</span> count_matrix_raw[</span>
<span id="cb50-207"><a href="#cb50-207" aria-hidden="true" tabindex="-1"></a>  <span class="sc">!</span><span class="fu">rownames</span>(count_matrix_raw) <span class="sc">%in%</span> genes_Y,</span>
<span id="cb50-208"><a href="#cb50-208" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb50-209"><a href="#cb50-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-210"><a href="#cb50-210" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-211"><a href="#cb50-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-212"><a href="#cb50-212" aria-hidden="true" tabindex="-1"></a><span class="fu">## Read and filter Normalized data</span></span>
<span id="cb50-213"><a href="#cb50-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-216"><a href="#cb50-216" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-217"><a href="#cb50-217" aria-hidden="true" tabindex="-1"></a>data_normalized <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="at">file =</span> <span class="st">"data/GSE260603_normalized_data.tsv"</span>)</span>
<span id="cb50-218"><a href="#cb50-218" aria-hidden="true" tabindex="-1"></a>data_normalized </span>
<span id="cb50-219"><a href="#cb50-219" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-220"><a href="#cb50-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-223"><a href="#cb50-223" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-224"><a href="#cb50-224" aria-hidden="true" tabindex="-1"></a><span class="co"># Transpose</span></span>
<span id="cb50-225"><a href="#cb50-225" aria-hidden="true" tabindex="-1"></a>df_t <span class="ot">&lt;-</span> <span class="fu">t</span>(data_normalized)</span>
<span id="cb50-226"><a href="#cb50-226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-227"><a href="#cb50-227" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to data frame</span></span>
<span id="cb50-228"><a href="#cb50-228" aria-hidden="true" tabindex="-1"></a>df_t <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(df_t)</span>
<span id="cb50-229"><a href="#cb50-229" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df_t) <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unlist</span>(df_t[<span class="dv">1</span>,]))</span>
<span id="cb50-230"><a href="#cb50-230" aria-hidden="true" tabindex="-1"></a>df_t<span class="ot">&lt;-</span>df_t[<span class="sc">-</span><span class="dv">1</span>,]</span>
<span id="cb50-231"><a href="#cb50-231" aria-hidden="true" tabindex="-1"></a>df_t</span>
<span id="cb50-232"><a href="#cb50-232" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-233"><a href="#cb50-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-236"><a href="#cb50-236" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-237"><a href="#cb50-237" aria-hidden="true" tabindex="-1"></a>expr <span class="ot">&lt;-</span> data_normalized <span class="sc">%&gt;%</span></span>
<span id="cb50-238"><a href="#cb50-238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb50-239"><a href="#cb50-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-240"><a href="#cb50-240" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(expr) <span class="ot">&lt;-</span> expr<span class="sc">$</span>Geneid</span>
<span id="cb50-241"><a href="#cb50-241" aria-hidden="true" tabindex="-1"></a>expr<span class="sc">$</span>Geneid <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb50-242"><a href="#cb50-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-243"><a href="#cb50-243" aria-hidden="true" tabindex="-1"></a>expr <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(expr)</span>
<span id="cb50-244"><a href="#cb50-244" aria-hidden="true" tabindex="-1"></a><span class="fu">storage.mode</span>(expr) <span class="ot">&lt;-</span> <span class="st">"numeric"</span></span>
<span id="cb50-245"><a href="#cb50-245" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(expr), <span class="at">center =</span> <span class="cn">TRUE</span>, <span class="at">scale. =</span> <span class="cn">TRUE</span>)</span>
<span id="cb50-246"><a href="#cb50-246" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-247"><a href="#cb50-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-250"><a href="#cb50-250" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-251"><a href="#cb50-251" aria-hidden="true" tabindex="-1"></a><span class="co"># data frame</span></span>
<span id="cb50-252"><a href="#cb50-252" aria-hidden="true" tabindex="-1"></a>pca_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pca<span class="sc">$</span>x) <span class="sc">%&gt;%</span></span>
<span id="cb50-253"><a href="#cb50-253" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">rownames</span>(.))</span>
<span id="cb50-254"><a href="#cb50-254" aria-hidden="true" tabindex="-1"></a>pca_df</span>
<span id="cb50-255"><a href="#cb50-255" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-256"><a href="#cb50-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-257"><a href="#cb50-257" aria-hidden="true" tabindex="-1"></a><span class="fu">## Inspect data</span></span>
<span id="cb50-258"><a href="#cb50-258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-261"><a href="#cb50-261" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-262"><a href="#cb50-262" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine PCA coordinates with metadata</span></span>
<span id="cb50-263"><a href="#cb50-263" aria-hidden="true" tabindex="-1"></a>pca_metadata_df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(pca_df, metadata_filt)</span>
<span id="cb50-264"><a href="#cb50-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-265"><a href="#cb50-265" aria-hidden="true" tabindex="-1"></a><span class="co"># pca gender</span></span>
<span id="cb50-266"><a href="#cb50-266" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_metadata_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> gender)) <span class="sc">+</span></span>
<span id="cb50-267"><a href="#cb50-267" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb50-268"><a href="#cb50-268" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Gender"</span>, <span class="at">color =</span> <span class="st">"Gender"</span>) <span class="sc">+</span></span>
<span id="cb50-269"><a href="#cb50-269" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>)</span>
<span id="cb50-270"><a href="#cb50-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-271"><a href="#cb50-271" aria-hidden="true" tabindex="-1"></a><span class="co">#pca affect</span></span>
<span id="cb50-272"><a href="#cb50-272" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_metadata_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> affected)) <span class="sc">+</span></span>
<span id="cb50-273"><a href="#cb50-273" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb50-274"><a href="#cb50-274" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Affected"</span>, <span class="at">color =</span> <span class="st">"Affected"</span>) <span class="sc">+</span></span>
<span id="cb50-275"><a href="#cb50-275" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>)</span>
<span id="cb50-276"><a href="#cb50-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-277"><a href="#cb50-277" aria-hidden="true" tabindex="-1"></a><span class="co">#pca bmi)</span></span>
<span id="cb50-278"><a href="#cb50-278" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_metadata_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> bmi)) <span class="sc">+</span></span>
<span id="cb50-279"><a href="#cb50-279" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb50-280"><a href="#cb50-280" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">"lightblue"</span>, <span class="at">high =</span> <span class="st">"darkblue"</span>) <span class="sc">+</span></span>
<span id="cb50-281"><a href="#cb50-281" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"BMI"</span>, <span class="at">color =</span> <span class="st">"BMI"</span>) <span class="sc">+</span></span>
<span id="cb50-282"><a href="#cb50-282" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>)</span>
<span id="cb50-283"><a href="#cb50-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-284"><a href="#cb50-284" aria-hidden="true" tabindex="-1"></a><span class="co"># pca age</span></span>
<span id="cb50-285"><a href="#cb50-285" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_metadata_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> age)) <span class="sc">+</span></span>
<span id="cb50-286"><a href="#cb50-286" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb50-287"><a href="#cb50-287" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">"lightgreen"</span>, <span class="at">high =</span> <span class="st">"darkgreen"</span>) <span class="sc">+</span></span>
<span id="cb50-288"><a href="#cb50-288" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Age"</span>, <span class="at">color =</span> <span class="st">"Age"</span>) <span class="sc">+</span></span>
<span id="cb50-289"><a href="#cb50-289" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>)</span>
<span id="cb50-290"><a href="#cb50-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-291"><a href="#cb50-291" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine 4 plots into 2x2 grid</span></span>
<span id="cb50-292"><a href="#cb50-292" aria-hidden="true" tabindex="-1"></a>(p1 <span class="sc">|</span> p2) <span class="sc">/</span> (p3 <span class="sc">|</span> p4)</span>
<span id="cb50-293"><a href="#cb50-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-294"><a href="#cb50-294" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-295"><a href="#cb50-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-296"><a href="#cb50-296" aria-hidden="true" tabindex="-1"></a><span class="fu">### Checkpoint</span></span>
<span id="cb50-297"><a href="#cb50-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-300"><a href="#cb50-300" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-301"><a href="#cb50-301" aria-hidden="true" tabindex="-1"></a><span class="co">#Save raw data</span></span>
<span id="cb50-302"><a href="#cb50-302" aria-hidden="true" tabindex="-1"></a><span class="co">#save(metadata, counts, file = "RData/input_raw.RData")</span></span>
<span id="cb50-303"><a href="#cb50-303" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-304"><a href="#cb50-304" aria-hidden="true" tabindex="-1"></a><span class="co">#SAVE DATA load for dds </span></span>
<span id="cb50-305"><a href="#cb50-305" aria-hidden="true" tabindex="-1"></a><span class="co">#save(metadata_filt, count_matrix, file = "RData/input_for_DGE_all.RData")</span></span>
<span id="cb50-306"><a href="#cb50-306" aria-hidden="true" tabindex="-1"></a><span class="co">#load("RData/input_for_DGE_all.RData")</span></span>
<span id="cb50-307"><a href="#cb50-307" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-308"><a href="#cb50-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-309"><a href="#cb50-309" aria-hidden="true" tabindex="-1"></a><span class="fu"># DGE DESeq2</span></span>
<span id="cb50-310"><a href="#cb50-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-313"><a href="#cb50-313" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-314"><a href="#cb50-314" aria-hidden="true" tabindex="-1"></a><span class="co">#Perform DGE</span></span>
<span id="cb50-315"><a href="#cb50-315" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(</span>
<span id="cb50-316"><a href="#cb50-316" aria-hidden="true" tabindex="-1"></a>  <span class="at">countData =</span> count_matrix,</span>
<span id="cb50-317"><a href="#cb50-317" aria-hidden="true" tabindex="-1"></a>  <span class="at">colData =</span> metadata_filt,</span>
<span id="cb50-318"><a href="#cb50-318" aria-hidden="true" tabindex="-1"></a>  <span class="at">design =</span> <span class="sc">~</span> affected <span class="sc">+</span> gender</span>
<span id="cb50-319"><a href="#cb50-319" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb50-320"><a href="#cb50-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-321"><a href="#cb50-321" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[<span class="fu">rowSums</span>(<span class="fu">counts</span>(dds)) <span class="sc">&gt;</span> <span class="dv">10</span>, ]</span>
<span id="cb50-322"><a href="#cb50-322" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb50-323"><a href="#cb50-323" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-324"><a href="#cb50-324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-327"><a href="#cb50-327" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-328"><a href="#cb50-328" aria-hidden="true" tabindex="-1"></a><span class="co">#Checkpoint</span></span>
<span id="cb50-329"><a href="#cb50-329" aria-hidden="true" tabindex="-1"></a><span class="co">#save(dds, file="RData/dds_all.Rdata")</span></span>
<span id="cb50-330"><a href="#cb50-330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-331"><a href="#cb50-331" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-332"><a href="#cb50-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-333"><a href="#cb50-333" aria-hidden="true" tabindex="-1"></a><span class="al">![]()</span>load("RData/dds_all.Rdata")</span>
<span id="cb50-334"><a href="#cb50-334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-337"><a href="#cb50-337" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-338"><a href="#cb50-338" aria-hidden="true" tabindex="-1"></a><span class="co">#Set pval and logfc</span></span>
<span id="cb50-339"><a href="#cb50-339" aria-hidden="true" tabindex="-1"></a>pval<span class="ot">=</span><span class="fl">0.01</span></span>
<span id="cb50-340"><a href="#cb50-340" aria-hidden="true" tabindex="-1"></a>logfc<span class="ot">=</span><span class="dv">1</span></span>
<span id="cb50-341"><a href="#cb50-341" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-342"><a href="#cb50-342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-345"><a href="#cb50-345" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-346"><a href="#cb50-346" aria-hidden="true" tabindex="-1"></a><span class="co">#Normalize data</span></span>
<span id="cb50-347"><a href="#cb50-347" aria-hidden="true" tabindex="-1"></a>norm_counts <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">counts</span>(dds, <span class="at">normalized =</span> T)</span>
<span id="cb50-348"><a href="#cb50-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-349"><a href="#cb50-349" aria-hidden="true" tabindex="-1"></a><span class="co"># Get result</span></span>
<span id="cb50-350"><a href="#cb50-350" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds)</span>
<span id="cb50-351"><a href="#cb50-351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-352"><a href="#cb50-352" aria-hidden="true" tabindex="-1"></a><span class="co">#Convert to data frame</span></span>
<span id="cb50-353"><a href="#cb50-353" aria-hidden="true" tabindex="-1"></a>dge_res_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(res)</span>
<span id="cb50-354"><a href="#cb50-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-355"><a href="#cb50-355" aria-hidden="true" tabindex="-1"></a><span class="co">#Drop row with Na in padj</span></span>
<span id="cb50-356"><a href="#cb50-356" aria-hidden="true" tabindex="-1"></a>dge_res_df <span class="ot">&lt;-</span><span class="fu">drop_na</span>(dge_res_df)</span>
<span id="cb50-357"><a href="#cb50-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-358"><a href="#cb50-358" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a column to the data frame to specify if they are UP- or DOWN- regulated (log2fc respectively positive or negative)</span></span>
<span id="cb50-359"><a href="#cb50-359" aria-hidden="true" tabindex="-1"></a>dge_res_df<span class="sc">$</span>diffexpressed <span class="ot">&lt;-</span> <span class="st">"NO"</span></span>
<span id="cb50-360"><a href="#cb50-360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-361"><a href="#cb50-361" aria-hidden="true" tabindex="-1"></a><span class="co"># Clasify gene using treshold</span></span>
<span id="cb50-362"><a href="#cb50-362" aria-hidden="true" tabindex="-1"></a>dge_res_df<span class="sc">$</span>diffexpressed[dge_res_df<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> logfc <span class="sc">&amp;</span> dge_res_df<span class="sc">$</span>pval <span class="sc">&lt;</span> pval] <span class="ot">&lt;-</span> <span class="st">"UP"</span></span>
<span id="cb50-363"><a href="#cb50-363" aria-hidden="true" tabindex="-1"></a>dge_res_df<span class="sc">$</span>diffexpressed[dge_res_df<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span>logfc <span class="sc">&amp;</span> dge_res_df<span class="sc">$</span>pval <span class="sc">&lt;</span> pval] <span class="ot">&lt;-</span> <span class="st">"DOWN"</span></span>
<span id="cb50-364"><a href="#cb50-364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-365"><a href="#cb50-365" aria-hidden="true" tabindex="-1"></a><span class="co">#Filter down genes</span></span>
<span id="cb50-366"><a href="#cb50-366" aria-hidden="true" tabindex="-1"></a>overexpressed_genes <span class="ot">&lt;-</span> <span class="fu">subset</span>(dge_res_df, diffexpressed<span class="sc">!=</span><span class="st">"NO"</span>)  </span>
<span id="cb50-367"><a href="#cb50-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-368"><a href="#cb50-368" aria-hidden="true" tabindex="-1"></a>dge_res_df</span>
<span id="cb50-369"><a href="#cb50-369" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-370"><a href="#cb50-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-373"><a href="#cb50-373" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-374"><a href="#cb50-374" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDispEsts</span>(dds)</span>
<span id="cb50-375"><a href="#cb50-375" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-376"><a href="#cb50-376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-377"><a href="#cb50-377" aria-hidden="true" tabindex="-1"></a><span class="fu">## Volcano Plot</span></span>
<span id="cb50-378"><a href="#cb50-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-381"><a href="#cb50-381" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-382"><a href="#cb50-382" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: VOlcano plot</span></span>
<span id="cb50-383"><a href="#cb50-383" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> dge_res_df, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">col =</span> diffexpressed)) <span class="sc">+</span></span>
<span id="cb50-384"><a href="#cb50-384" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(<span class="sc">-</span>logfc,logfc), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span></span>
<span id="cb50-385"><a href="#cb50-385" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">col =</span> <span class="st">"gray"</span>, <span class="at">linetype =</span> <span class="st">'dashed'</span>) <span class="sc">+</span> </span>
<span id="cb50-386"><a href="#cb50-386" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb50-387"><a href="#cb50-387" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#00AFBB"</span>, <span class="st">"grey"</span>, <span class="st">"#bb0c00"</span>), </span>
<span id="cb50-388"><a href="#cb50-388" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Downregulated"</span>, <span class="st">"Not significant"</span>, <span class="st">"Upregulated"</span>)) <span class="sc">+</span> </span>
<span id="cb50-389"><a href="#cb50-389" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">200</span>), <span class="at">xlim =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="dv">10</span>)) <span class="sc">+</span> </span>
<span id="cb50-390"><a href="#cb50-390" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">color =</span> <span class="st">'Genes'</span>, <span class="co">#legend_title, </span></span>
<span id="cb50-391"><a href="#cb50-391" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"log"</span>[<span class="dv">2</span>]<span class="sc">*</span><span class="st">"FC"</span>), </span>
<span id="cb50-392"><a href="#cb50-392" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"-log"</span>[<span class="dv">10</span>]<span class="sc">*</span><span class="st">"p-adj"</span>)) <span class="sc">+</span> </span>
<span id="cb50-393"><a href="#cb50-393" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">10</span>, <span class="dv">10</span>, <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb50-394"><a href="#cb50-394" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">'DGE bet'</span>) </span>
<span id="cb50-395"><a href="#cb50-395" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb50-396"><a href="#cb50-396" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-397"><a href="#cb50-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-398"><a href="#cb50-398" aria-hidden="true" tabindex="-1"></a><span class="al">![]()</span></span>
<span id="cb50-399"><a href="#cb50-399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-400"><a href="#cb50-400" aria-hidden="true" tabindex="-1"></a><span class="fu"># PCA</span></span>
<span id="cb50-401"><a href="#cb50-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-404"><a href="#cb50-404" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-405"><a href="#cb50-405" aria-hidden="true" tabindex="-1"></a><span class="co"># transpose a table</span></span>
<span id="cb50-406"><a href="#cb50-406" aria-hidden="true" tabindex="-1"></a>to_PCA <span class="ot">&lt;-</span> norm_counts <span class="sc">%&gt;%</span> <span class="fu">t</span>()  </span>
<span id="cb50-407"><a href="#cb50-407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-408"><a href="#cb50-408" aria-hidden="true" tabindex="-1"></a><span class="co">#Run PCA</span></span>
<span id="cb50-409"><a href="#cb50-409" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(to_PCA) <span class="co">#run PCA (the result is a list)</span></span>
<span id="cb50-410"><a href="#cb50-410" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-411"><a href="#cb50-411" aria-hidden="true" tabindex="-1"></a><span class="co">#Transform to dataframe</span></span>
<span id="cb50-412"><a href="#cb50-412" aria-hidden="true" tabindex="-1"></a><span class="co">#Tu się coś wykrzacza ale wystarzy pujśc dalej i działa xd</span></span>
<span id="cb50-413"><a href="#cb50-413" aria-hidden="true" tabindex="-1"></a>pca_plot <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pca<span class="sc">$</span>x) </span>
<span id="cb50-414"><a href="#cb50-414" aria-hidden="true" tabindex="-1"></a>pca_plot</span>
<span id="cb50-415"><a href="#cb50-415" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-416"><a href="#cb50-416" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-419"><a href="#cb50-419" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-420"><a href="#cb50-420" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: PCA_1</span></span>
<span id="cb50-421"><a href="#cb50-421" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine PCA coordinates with metadata</span></span>
<span id="cb50-422"><a href="#cb50-422" aria-hidden="true" tabindex="-1"></a>pca_metadata_df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(pca_plot, metadata_filt)</span>
<span id="cb50-423"><a href="#cb50-423" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-424"><a href="#cb50-424" aria-hidden="true" tabindex="-1"></a><span class="co"># pca gender</span></span>
<span id="cb50-425"><a href="#cb50-425" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_metadata_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> gender)) <span class="sc">+</span></span>
<span id="cb50-426"><a href="#cb50-426" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb50-427"><a href="#cb50-427" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Gender"</span>, <span class="at">color =</span> <span class="st">"Gender"</span>) <span class="sc">+</span></span>
<span id="cb50-428"><a href="#cb50-428" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>)</span>
<span id="cb50-429"><a href="#cb50-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-430"><a href="#cb50-430" aria-hidden="true" tabindex="-1"></a><span class="co">#pca affect</span></span>
<span id="cb50-431"><a href="#cb50-431" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_metadata_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> affected)) <span class="sc">+</span></span>
<span id="cb50-432"><a href="#cb50-432" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb50-433"><a href="#cb50-433" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Affected"</span>, <span class="at">color =</span> <span class="st">"Affected"</span>) <span class="sc">+</span></span>
<span id="cb50-434"><a href="#cb50-434" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>)</span>
<span id="cb50-435"><a href="#cb50-435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-436"><a href="#cb50-436" aria-hidden="true" tabindex="-1"></a><span class="co">#pca bmi)</span></span>
<span id="cb50-437"><a href="#cb50-437" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_metadata_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> bmi)) <span class="sc">+</span></span>
<span id="cb50-438"><a href="#cb50-438" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb50-439"><a href="#cb50-439" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">"lightblue"</span>, <span class="at">high =</span> <span class="st">"darkblue"</span>) <span class="sc">+</span></span>
<span id="cb50-440"><a href="#cb50-440" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"BMI"</span>, <span class="at">color =</span> <span class="st">"BMI"</span>) <span class="sc">+</span></span>
<span id="cb50-441"><a href="#cb50-441" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>)</span>
<span id="cb50-442"><a href="#cb50-442" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-443"><a href="#cb50-443" aria-hidden="true" tabindex="-1"></a><span class="co"># pca age</span></span>
<span id="cb50-444"><a href="#cb50-444" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_metadata_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> age)) <span class="sc">+</span></span>
<span id="cb50-445"><a href="#cb50-445" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb50-446"><a href="#cb50-446" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">"lightgreen"</span>, <span class="at">high =</span> <span class="st">"darkgreen"</span>) <span class="sc">+</span></span>
<span id="cb50-447"><a href="#cb50-447" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Age"</span>, <span class="at">color =</span> <span class="st">"Age"</span>) <span class="sc">+</span></span>
<span id="cb50-448"><a href="#cb50-448" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">12</span>)</span>
<span id="cb50-449"><a href="#cb50-449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-450"><a href="#cb50-450" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine 4 plots into 2x2 grid</span></span>
<span id="cb50-451"><a href="#cb50-451" aria-hidden="true" tabindex="-1"></a>(p1 <span class="sc">|</span> p2) <span class="sc">/</span> (p3 <span class="sc">|</span> p4)</span>
<span id="cb50-452"><a href="#cb50-452" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-453"><a href="#cb50-453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-454"><a href="#cb50-454" aria-hidden="true" tabindex="-1"></a><span class="al">![]()</span></span>
<span id="cb50-455"><a href="#cb50-455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-456"><a href="#cb50-456" aria-hidden="true" tabindex="-1"></a><span class="fu"># Gene function</span></span>
<span id="cb50-457"><a href="#cb50-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-460"><a href="#cb50-460" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-461"><a href="#cb50-461" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(msigdbr)</span>
<span id="cb50-462"><a href="#cb50-462" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb50-463"><a href="#cb50-463" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-464"><a href="#cb50-464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-467"><a href="#cb50-467" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-468"><a href="#cb50-468" aria-hidden="true" tabindex="-1"></a>dge_res_df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/dge_res_df.csv"</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb50-469"><a href="#cb50-469" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-470"><a href="#cb50-470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-473"><a href="#cb50-473" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-474"><a href="#cb50-474" aria-hidden="true" tabindex="-1"></a>name_of_gene_set <span class="ot">&lt;-</span> <span class="st">'C7'</span></span>
<span id="cb50-475"><a href="#cb50-475" aria-hidden="true" tabindex="-1"></a>gene_sets_df <span class="ot">&lt;-</span> <span class="fu">msigdbr</span>(<span class="at">species =</span> <span class="st">"Homo sapiens"</span>, <span class="at">category =</span> name_of_gene_set)</span>
<span id="cb50-476"><a href="#cb50-476" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gene_sets_df)</span>
<span id="cb50-477"><a href="#cb50-477" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-478"><a href="#cb50-478" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-481"><a href="#cb50-481" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-482"><a href="#cb50-482" aria-hidden="true" tabindex="-1"></a>gene_sets <span class="ot">&lt;-</span> gene_sets_df <span class="sc">%&gt;%</span></span>
<span id="cb50-483"><a href="#cb50-483" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gs_name, ensembl_gene)</span>
<span id="cb50-484"><a href="#cb50-484" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-485"><a href="#cb50-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-488"><a href="#cb50-488" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-489"><a href="#cb50-489" aria-hidden="true" tabindex="-1"></a>df_sig <span class="ot">&lt;-</span> dge_res_df[<span class="fu">which</span>(dge_res_df<span class="sc">$</span>diffexpressed<span class="sc">!=</span><span class="st">"NO"</span>),]</span>
<span id="cb50-490"><a href="#cb50-490" aria-hidden="true" tabindex="-1"></a>deg_results_list <span class="ot">&lt;-</span> <span class="fu">split</span>(df_sig, df_sig<span class="sc">$</span>diffexpressed)</span>
<span id="cb50-491"><a href="#cb50-491" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-492"><a href="#cb50-492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-495"><a href="#cb50-495" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-496"><a href="#cb50-496" aria-hidden="true" tabindex="-1"></a>background_genes <span class="ot">&lt;-</span> data_normalized <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(Geneid)</span>
<span id="cb50-497"><a href="#cb50-497" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-498"><a href="#cb50-498" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-501"><a href="#cb50-501" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-502"><a href="#cb50-502" aria-hidden="true" tabindex="-1"></a><span class="do">## Run ClusterProfiler</span></span>
<span id="cb50-503"><a href="#cb50-503" aria-hidden="true" tabindex="-1"></a><span class="co"># Settings</span></span>
<span id="cb50-504"><a href="#cb50-504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-505"><a href="#cb50-505" aria-hidden="true" tabindex="-1"></a>name_of_comparison <span class="ot">&lt;-</span> <span class="st">'severevshealthy'</span></span>
<span id="cb50-506"><a href="#cb50-506" aria-hidden="true" tabindex="-1"></a>padj_cutoff <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb50-507"><a href="#cb50-507" aria-hidden="true" tabindex="-1"></a>genecount_cutoff <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb50-508"><a href="#cb50-508" aria-hidden="true" tabindex="-1"></a>filename_prefix <span class="ot">&lt;-</span> <span class="fu">paste0</span>(name_of_comparison, <span class="st">'_'</span>, name_of_gene_set)</span>
<span id="cb50-509"><a href="#cb50-509" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-510"><a href="#cb50-510" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-513"><a href="#cb50-513" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-514"><a href="#cb50-514" aria-hidden="true" tabindex="-1"></a><span class="co"># Run clusterProfiler on each sub-dataframe</span></span>
<span id="cb50-515"><a href="#cb50-515" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">names</span>(deg_results_list),</span>
<span id="cb50-516"><a href="#cb50-516" aria-hidden="true" tabindex="-1"></a>              <span class="cf">function</span>(x) <span class="fu">enricher</span>(<span class="at">gene =</span> <span class="fu">rownames</span>(deg_results_list[[x]]),</span>
<span id="cb50-517"><a href="#cb50-517" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">TERM2GENE =</span> gene_sets,</span>
<span id="cb50-518"><a href="#cb50-518" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">universe =</span> background_genes,</span>
<span id="cb50-519"><a href="#cb50-519" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">qvalueCutoff =</span> <span class="fl">0.2</span>))</span>
<span id="cb50-520"><a href="#cb50-520" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(res) <span class="ot">&lt;-</span> <span class="fu">names</span>(deg_results_list)</span>
<span id="cb50-521"><a href="#cb50-521" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-522"><a href="#cb50-522" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"Enrichment analysis results:</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb50-523"><a href="#cb50-523" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">names</span>(res)) {</span>
<span id="cb50-524"><a href="#cb50-524" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">paste</span>(i, <span class="st">":"</span>, <span class="fu">nrow</span>(<span class="fu">as.data.frame</span>(res[[i]])), <span class="st">"enriched pathways</span><span class="sc">\n</span><span class="st">"</span>))</span>
<span id="cb50-525"><a href="#cb50-525" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb50-526"><a href="#cb50-526" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-527"><a href="#cb50-527" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-530"><a href="#cb50-530" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-531"><a href="#cb50-531" aria-hidden="true" tabindex="-1"></a><span class="co">#Convert the enrichResults to a dataframe with the pathways</span></span>
<span id="cb50-532"><a href="#cb50-532" aria-hidden="true" tabindex="-1"></a>res_df <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">names</span>(res), <span class="cf">function</span>(x) <span class="fu">rbind</span>(res[[x]]<span class="sc">@</span>result))</span>
<span id="cb50-533"><a href="#cb50-533" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(res_df) <span class="ot">&lt;-</span> <span class="fu">names</span>(res)</span>
<span id="cb50-534"><a href="#cb50-534" aria-hidden="true" tabindex="-1"></a>res_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, res_df)</span>
<span id="cb50-535"><a href="#cb50-535" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_df)</span>
<span id="cb50-536"><a href="#cb50-536" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-537"><a href="#cb50-537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-540"><a href="#cb50-540" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-541"><a href="#cb50-541" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(res_df, <span class="st">"results.csv"</span>)</span>
<span id="cb50-542"><a href="#cb50-542" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-543"><a href="#cb50-543" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-544"><a href="#cb50-544" aria-hidden="true" tabindex="-1"></a>Differential expression analysis revealed that the upregulated gene set was dominated by non-coding transcripts, including long non-coding RNAs (lncRNAs), Y RNAs, and several transcribed pseudogenes (e.g. PRKY, ANOS2P, TXLNGY). Most of these transcripts are poorly characterized and lack evidence for protein-coding potential or involvement in known biological pathways. In contrast, the downregulated genes included protein-coding genes associated with cytoskeletal organization (MAP7D2), regulation of protein phosphatase activity (PPP1R2C), and cell cycle–related processes such as mitotic spindle control and chromatid segregation (ERCC6L). Additionally, two key long non-coding RNAs involved in X-chromosome inactivation, TSIX and XIST, were significantly downregulated.</span>
<span id="cb50-545"><a href="#cb50-545" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-546"><a href="#cb50-546" aria-hidden="true" tabindex="-1"></a><span class="fu"># Multiomics clustering SUMO</span></span>
<span id="cb50-547"><a href="#cb50-547" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-550"><a href="#cb50-550" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-551"><a href="#cb50-551" aria-hidden="true" tabindex="-1"></a>imune_marker_raw<span class="ot">=</span><span class="fu">read.csv</span>(<span class="st">"data/immune_marker_data.tsv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb50-552"><a href="#cb50-552" aria-hidden="true" tabindex="-1"></a>imune_marker_raw</span>
<span id="cb50-553"><a href="#cb50-553" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-554"><a href="#cb50-554" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-557"><a href="#cb50-557" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-558"><a href="#cb50-558" aria-hidden="true" tabindex="-1"></a><span class="co">#Filter out not use sample</span></span>
<span id="cb50-559"><a href="#cb50-559" aria-hidden="true" tabindex="-1"></a>imune_marker<span class="ot">&lt;-</span> imune_marker_raw <span class="sc">%&gt;%</span> <span class="fu">filter</span>(analysis_type<span class="sc">==</span><span class="st">"initial"</span>) <span class="sc">%&gt;%</span> <span class="fu">slice</span> (<span class="dv">1</span><span class="sc">:</span><span class="dv">262</span>)</span>
<span id="cb50-560"><a href="#cb50-560" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-561"><a href="#cb50-561" aria-hidden="true" tabindex="-1"></a><span class="co">#Change rownames</span></span>
<span id="cb50-562"><a href="#cb50-562" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(imune_marker) <span class="ot">&lt;-</span> imune_marker<span class="sc">$</span>ID</span>
<span id="cb50-563"><a href="#cb50-563" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-564"><a href="#cb50-564" aria-hidden="true" tabindex="-1"></a><span class="co">#Save to file</span></span>
<span id="cb50-565"><a href="#cb50-565" aria-hidden="true" tabindex="-1"></a>to_save <span class="ot">&lt;-</span> imune_marker <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(age, sex, BMI, status, analysis_type, batch_variable))</span>
<span id="cb50-566"><a href="#cb50-566" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(to_save, <span class="at">file =</span> <span class="st">"sumo/immune_markers.tsv"</span>)</span>
<span id="cb50-567"><a href="#cb50-567" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-568"><a href="#cb50-568" aria-hidden="true" tabindex="-1"></a>imune_marker_numeric <span class="ot">&lt;-</span> imune_marker <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">!</span><span class="fu">c</span>(ID, age, sex, BMI, status, analysis_type, batch_variable))</span>
<span id="cb50-569"><a href="#cb50-569" aria-hidden="true" tabindex="-1"></a>imune_marker_numeric</span>
<span id="cb50-570"><a href="#cb50-570" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-571"><a href="#cb50-571" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-572"><a href="#cb50-572" aria-hidden="true" tabindex="-1"></a><span class="fu">## Normalize data</span></span>
<span id="cb50-573"><a href="#cb50-573" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-576"><a href="#cb50-576" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-577"><a href="#cb50-577" aria-hidden="true" tabindex="-1"></a><span class="co">#Prepare data for sumo</span></span>
<span id="cb50-578"><a href="#cb50-578" aria-hidden="true" tabindex="-1"></a><span class="co">#Handle missing values, fill NA with mean()</span></span>
<span id="cb50-579"><a href="#cb50-579" aria-hidden="true" tabindex="-1"></a>df_imput <span class="ot">&lt;-</span> (</span>
<span id="cb50-580"><a href="#cb50-580" aria-hidden="true" tabindex="-1"></a>  <span class="fu">apply</span>(imune_marker_numeric, <span class="dv">1</span>, <span class="cf">function</span>(x) {</span>
<span id="cb50-581"><a href="#cb50-581" aria-hidden="true" tabindex="-1"></a>    x[<span class="fu">is.na</span>(x)] <span class="ot">&lt;-</span> <span class="fu">mean</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb50-582"><a href="#cb50-582" aria-hidden="true" tabindex="-1"></a>    x</span>
<span id="cb50-583"><a href="#cb50-583" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb50-584"><a href="#cb50-584" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb50-585"><a href="#cb50-585" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-586"><a href="#cb50-586" aria-hidden="true" tabindex="-1"></a><span class="co">#Feature-wise standardization (mean=0, sd=1)</span></span>
<span id="cb50-587"><a href="#cb50-587" aria-hidden="true" tabindex="-1"></a>df_scaled <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(df_imput)))</span>
<span id="cb50-588"><a href="#cb50-588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-589"><a href="#cb50-589" aria-hidden="true" tabindex="-1"></a><span class="co">#Verify scaling</span></span>
<span id="cb50-590"><a href="#cb50-590" aria-hidden="true" tabindex="-1"></a>feature_means <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(df_scaled)</span>
<span id="cb50-591"><a href="#cb50-591" aria-hidden="true" tabindex="-1"></a>feature_sds <span class="ot">&lt;-</span> <span class="fu">apply</span>(df_scaled, <span class="dv">1</span>, sd)</span>
<span id="cb50-592"><a href="#cb50-592" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-593"><a href="#cb50-593" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(feature_means)</span>
<span id="cb50-594"><a href="#cb50-594" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(feature_sds)</span>
<span id="cb50-595"><a href="#cb50-595" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-596"><a href="#cb50-596" aria-hidden="true" tabindex="-1"></a><span class="co"># Write output</span></span>
<span id="cb50-597"><a href="#cb50-597" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(</span>
<span id="cb50-598"><a href="#cb50-598" aria-hidden="true" tabindex="-1"></a>  df_scaled,</span>
<span id="cb50-599"><a href="#cb50-599" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"sumo/immune_markers.tsv"</span>,</span>
<span id="cb50-600"><a href="#cb50-600" aria-hidden="true" tabindex="-1"></a>  <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb50-601"><a href="#cb50-601" aria-hidden="true" tabindex="-1"></a>  <span class="at">quote =</span> <span class="cn">FALSE</span>,</span>
<span id="cb50-602"><a href="#cb50-602" aria-hidden="true" tabindex="-1"></a>  <span class="at">col.names =</span> <span class="cn">NA</span></span>
<span id="cb50-603"><a href="#cb50-603" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb50-604"><a href="#cb50-604" aria-hidden="true" tabindex="-1"></a>t</span>
<span id="cb50-605"><a href="#cb50-605" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-606"><a href="#cb50-606" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-607"><a href="#cb50-607" aria-hidden="true" tabindex="-1"></a><span class="fu">### DGE data</span></span>
<span id="cb50-608"><a href="#cb50-608" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-611"><a href="#cb50-611" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-612"><a href="#cb50-612" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb50-613"><a href="#cb50-613" aria-hidden="true" tabindex="-1"></a>expr <span class="ot">&lt;-</span> <span class="fu">assay</span>(vsd)</span>
<span id="cb50-614"><a href="#cb50-614" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-615"><a href="#cb50-615" aria-hidden="true" tabindex="-1"></a><span class="co"># Variable genes (recommended)</span></span>
<span id="cb50-616"><a href="#cb50-616" aria-hidden="true" tabindex="-1"></a>vars <span class="ot">&lt;-</span> <span class="fu">apply</span>(expr, <span class="dv">1</span>, var)</span>
<span id="cb50-617"><a href="#cb50-617" aria-hidden="true" tabindex="-1"></a>genes_use <span class="ot">&lt;-</span> <span class="fu">names</span>(<span class="fu">sort</span>(vars, <span class="at">decreasing =</span> <span class="cn">TRUE</span>))[<span class="dv">1</span><span class="sc">:</span><span class="dv">2000</span>]</span>
<span id="cb50-618"><a href="#cb50-618" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-619"><a href="#cb50-619" aria-hidden="true" tabindex="-1"></a>expr <span class="ot">&lt;-</span> expr[genes_use, ]</span>
<span id="cb50-620"><a href="#cb50-620" aria-hidden="true" tabindex="-1"></a>expr_scaled <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(expr)))</span>
<span id="cb50-621"><a href="#cb50-621" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-622"><a href="#cb50-622" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(</span>
<span id="cb50-623"><a href="#cb50-623" aria-hidden="true" tabindex="-1"></a>  expr_scaled,</span>
<span id="cb50-624"><a href="#cb50-624" aria-hidden="true" tabindex="-1"></a>  <span class="st">"sumo/sumo_rnaseq.tsv"</span>,</span>
<span id="cb50-625"><a href="#cb50-625" aria-hidden="true" tabindex="-1"></a>  <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,</span>
<span id="cb50-626"><a href="#cb50-626" aria-hidden="true" tabindex="-1"></a>  <span class="at">quote =</span> <span class="cn">FALSE</span>,</span>
<span id="cb50-627"><a href="#cb50-627" aria-hidden="true" tabindex="-1"></a>  <span class="at">col.names =</span> <span class="cn">NA</span></span>
<span id="cb50-628"><a href="#cb50-628" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb50-629"><a href="#cb50-629" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-630"><a href="#cb50-630" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-631"><a href="#cb50-631" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-632"><a href="#cb50-632" aria-hidden="true" tabindex="-1"></a><span class="fu">### Metadata</span></span>
<span id="cb50-633"><a href="#cb50-633" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-636"><a href="#cb50-636" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-637"><a href="#cb50-637" aria-hidden="true" tabindex="-1"></a><span class="co">#Create labels.tsv We need becouse of sumo evaluate</span></span>
<span id="cb50-638"><a href="#cb50-638" aria-hidden="true" tabindex="-1"></a>label_affected <span class="ot">&lt;-</span> metadata_filt <span class="sc">%&gt;%</span></span>
<span id="cb50-639"><a href="#cb50-639" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-640"><a href="#cb50-640" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">case_when</span>(</span>
<span id="cb50-641"><a href="#cb50-641" aria-hidden="true" tabindex="-1"></a>    affected <span class="sc">==</span> <span class="st">"control"</span>  <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb50-642"><a href="#cb50-642" aria-hidden="true" tabindex="-1"></a>    affected <span class="sc">==</span> <span class="st">"affected"</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb50-643"><a href="#cb50-643" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span>                   <span class="sc">~</span> <span class="cn">NA_real_</span></span>
<span id="cb50-644"><a href="#cb50-644" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span></span>
<span id="cb50-645"><a href="#cb50-645" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, label) <span class="sc">%&gt;%</span></span>
<span id="cb50-646"><a href="#cb50-646" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb50-647"><a href="#cb50-647" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-648"><a href="#cb50-648" aria-hidden="true" tabindex="-1"></a>label_gender <span class="ot">&lt;-</span> metadata_filt <span class="sc">%&gt;%</span></span>
<span id="cb50-649"><a href="#cb50-649" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-650"><a href="#cb50-650" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">case_when</span>(</span>
<span id="cb50-651"><a href="#cb50-651" aria-hidden="true" tabindex="-1"></a>    gender <span class="sc">==</span> <span class="st">"male"</span>  <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb50-652"><a href="#cb50-652" aria-hidden="true" tabindex="-1"></a>    gender <span class="sc">==</span> <span class="st">"female"</span> <span class="sc">~</span> <span class="dv">1</span>,</span>
<span id="cb50-653"><a href="#cb50-653" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span>                   <span class="sc">~</span> <span class="cn">NA_real_</span></span>
<span id="cb50-654"><a href="#cb50-654" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span></span>
<span id="cb50-655"><a href="#cb50-655" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, label) <span class="sc">%&gt;%</span></span>
<span id="cb50-656"><a href="#cb50-656" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb50-657"><a href="#cb50-657" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-658"><a href="#cb50-658" aria-hidden="true" tabindex="-1"></a>label_affected</span>
<span id="cb50-659"><a href="#cb50-659" aria-hidden="true" tabindex="-1"></a>label_gender</span>
<span id="cb50-660"><a href="#cb50-660" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-661"><a href="#cb50-661" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(label_affected, <span class="st">"sumo/labels_affected.tsv"</span>)</span>
<span id="cb50-662"><a href="#cb50-662" aria-hidden="true" tabindex="-1"></a><span class="fu">write_tsv</span>(label_gender, <span class="st">"sumo/labels_gender.tsv"</span>)</span>
<span id="cb50-663"><a href="#cb50-663" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-664"><a href="#cb50-664" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-665"><a href="#cb50-665" aria-hidden="true" tabindex="-1"></a><span class="fu">## Run SUMO</span></span>
<span id="cb50-666"><a href="#cb50-666" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-669"><a href="#cb50-669" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb50-670"><a href="#cb50-670" aria-hidden="true" tabindex="-1"></a><span class="co">#source /home/janek/miniconda3/bin/activate #required by Quarto to recognize conda's commands</span></span>
<span id="cb50-671"><a href="#cb50-671" aria-hidden="true" tabindex="-1"></a><span class="co">#conda activate SUMO</span></span>
<span id="cb50-672"><a href="#cb50-672" aria-hidden="true" tabindex="-1"></a><span class="co">#cd sumo</span></span>
<span id="cb50-673"><a href="#cb50-673" aria-hidden="true" tabindex="-1"></a><span class="co">#sumo prepare -atol 0.1 -plot LAML.png immune_markers.tsv,sumo_rnaseq.tsv output.npz </span></span>
<span id="cb50-674"><a href="#cb50-674" aria-hidden="true" tabindex="-1"></a><span class="co">#cd ..</span></span>
<span id="cb50-675"><a href="#cb50-675" aria-hidden="true" tabindex="-1"></a><span class="co">#conda deactivate #deactivate the environment</span></span>
<span id="cb50-676"><a href="#cb50-676" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-677"><a href="#cb50-677" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-680"><a href="#cb50-680" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb50-681"><a href="#cb50-681" aria-hidden="true" tabindex="-1"></a><span class="co">#source /home/janek/miniconda3/bin/activate #required by Quarto to recognize conda's commands</span></span>
<span id="cb50-682"><a href="#cb50-682" aria-hidden="true" tabindex="-1"></a><span class="co">#conda activate SUMO</span></span>
<span id="cb50-683"><a href="#cb50-683" aria-hidden="true" tabindex="-1"></a><span class="co">#cd sumo</span></span>
<span id="cb50-684"><a href="#cb50-684" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-685"><a href="#cb50-685" aria-hidden="true" tabindex="-1"></a><span class="co">#sumo run output.npz 2,4 LAML</span></span>
<span id="cb50-686"><a href="#cb50-686" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-687"><a href="#cb50-687" aria-hidden="true" tabindex="-1"></a><span class="co">#cd ..</span></span>
<span id="cb50-688"><a href="#cb50-688" aria-hidden="true" tabindex="-1"></a><span class="co">#conda deactivate #deactivate the environment</span></span>
<span id="cb50-689"><a href="#cb50-689" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-690"><a href="#cb50-690" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-693"><a href="#cb50-693" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb50-694"><a href="#cb50-694" aria-hidden="true" tabindex="-1"></a><span class="co">#source /home/janek/miniconda3/bin/activate #required by Quarto to recognize conda's commands</span></span>
<span id="cb50-695"><a href="#cb50-695" aria-hidden="true" tabindex="-1"></a><span class="co">#conda activate SUMO</span></span>
<span id="cb50-696"><a href="#cb50-696" aria-hidden="true" tabindex="-1"></a><span class="co">#cd sumo</span></span>
<span id="cb50-697"><a href="#cb50-697" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-698"><a href="#cb50-698" aria-hidden="true" tabindex="-1"></a><span class="co">#sumo run output.npz 2,7 LAML -rep 10 -r 4 </span></span>
<span id="cb50-699"><a href="#cb50-699" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-700"><a href="#cb50-700" aria-hidden="true" tabindex="-1"></a><span class="co">#cd ..</span></span>
<span id="cb50-701"><a href="#cb50-701" aria-hidden="true" tabindex="-1"></a><span class="co">#conda deactivate #deactivate the environment</span></span>
<span id="cb50-702"><a href="#cb50-702" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-703"><a href="#cb50-703" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-706"><a href="#cb50-706" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb50-707"><a href="#cb50-707" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> /home/janek/miniconda3/bin/activate <span class="co">#required by Quarto to recognize conda's commands</span></span>
<span id="cb50-708"><a href="#cb50-708" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate SUMO</span>
<span id="cb50-709"><a href="#cb50-709" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> sumo</span>
<span id="cb50-710"><a href="#cb50-710" aria-hidden="true" tabindex="-1"></a><span class="ex">sumo</span> evaluate LAML/k2/clusters.tsv labels_affected.tsv</span>
<span id="cb50-711"><a href="#cb50-711" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ..</span>
<span id="cb50-712"><a href="#cb50-712" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate <span class="co">#deactivate the environment</span></span>
<span id="cb50-713"><a href="#cb50-713" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-714"><a href="#cb50-714" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-715"><a href="#cb50-715" aria-hidden="true" tabindex="-1"></a>How to inerprate? NMI (Normalized Mutual Information):</span>
<span id="cb50-716"><a href="#cb50-716" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-717"><a href="#cb50-717" aria-hidden="true" tabindex="-1"></a><span class="fu">## Anylise SUMO output</span></span>
<span id="cb50-718"><a href="#cb50-718" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-721"><a href="#cb50-721" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-722"><a href="#cb50-722" aria-hidden="true" tabindex="-1"></a><span class="co"># Cluster assignments</span></span>
<span id="cb50-723"><a href="#cb50-723" aria-hidden="true" tabindex="-1"></a>clusters <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"sumo/LAML/k2/clusters.tsv"</span>, <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb50-724"><a href="#cb50-724" aria-hidden="true" tabindex="-1"></a>clusters</span>
<span id="cb50-725"><a href="#cb50-725" aria-hidden="true" tabindex="-1"></a><span class="co"># columns: Sample, Cluster</span></span>
<span id="cb50-726"><a href="#cb50-726" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(clusters)<span class="ot">&lt;-</span>clusters<span class="sc">$</span>sample</span>
<span id="cb50-727"><a href="#cb50-727" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge clusters with metadata</span></span>
<span id="cb50-728"><a href="#cb50-728" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">merge</span>(clusters, metadata_filt, <span class="at">by=</span><span class="st">'row.names'</span>)</span>
<span id="cb50-729"><a href="#cb50-729" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(df)</span>
<span id="cb50-730"><a href="#cb50-730" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-731"><a href="#cb50-731" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-734"><a href="#cb50-734" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-735"><a href="#cb50-735" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(df<span class="sc">$</span>label, df<span class="sc">$</span>affected)</span>
<span id="cb50-736"><a href="#cb50-736" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-737"><a href="#cb50-737" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-738"><a href="#cb50-738" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-741"><a href="#cb50-741" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-742"><a href="#cb50-742" aria-hidden="true" tabindex="-1"></a><span class="fu">prop.table</span>(<span class="fu">table</span>(df<span class="sc">$</span>label, df<span class="sc">$</span>affected), <span class="dv">1</span>)</span>
<span id="cb50-743"><a href="#cb50-743" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-744"><a href="#cb50-744" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-745"><a href="#cb50-745" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-748"><a href="#cb50-748" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-749"><a href="#cb50-749" aria-hidden="true" tabindex="-1"></a><span class="fu">fisher.test</span>(<span class="fu">table</span>(df<span class="sc">$</span>label, df<span class="sc">$</span>affected))</span>
<span id="cb50-750"><a href="#cb50-750" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-751"><a href="#cb50-751" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb50-752"><a href="#cb50-752" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-755"><a href="#cb50-755" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb50-756"><a href="#cb50-756" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">factor</span>(label), <span class="at">fill=</span><span class="fu">factor</span>(affected))) <span class="sc">+</span></span>
<span id="cb50-757"><a href="#cb50-757" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">position=</span><span class="st">"fill"</span>) <span class="sc">+</span></span>
<span id="cb50-758"><a href="#cb50-758" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Cluster"</span>, <span class="at">y=</span><span class="st">"Proportion"</span>, <span class="at">fill=</span><span class="st">"Affected"</span>) <span class="sc">+</span></span>
<span id="cb50-759"><a href="#cb50-759" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels=</span>scales<span class="sc">::</span><span class="fu">percent_format</span>()) <span class="sc">+</span></span>
<span id="cb50-760"><a href="#cb50-760" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span>
<span id="cb50-761"><a href="#cb50-761" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-762"><a href="#cb50-762" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code></pre></div><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>