---
title: "Untitled"
format: html
editor: visual
---

## Quarto

Quarto enables you to weave together content and executable code into a finished document. To learn more about Quarto see <https://quarto.org>.

## Introduction
The contemporary world, with its financial inequalities, employment instability, and global crises such as climate change or the pandemic, creates a favorable environment for the development of affective disorders (Ridley et al., 2020). Within just the first two decades of the 21st century, the number of people suffering from depression increased by nearly 60%, underscoring the need for a deeper understanding of this disorder and its etiology in order to effectively address the growing mental health crisis (Abbafati et al., 2020).

According to the ICD-11 classification, the diagnosis of a depressive episode requires the presence of at least five symptoms, one of which must belong to the affective cluster (Gałecki & Szulc, 2023). Additional symptoms include difficulties with concentration and memory, extremely low self-esteem, inappropriate feelings of guilt, sleep and appetite disturbances, as well as suicidal thoughts or behaviors. The ICD-11 classification also allows for the specification of symptom severity and the presence of psychotic features, which is crucial when deciding whether pharmacological treatment should be initiated.

The inflammatory theory of depression postulates that chronic inflammation in the body affects brain function, leading to the development of depressive symptoms. In individuals with depression, elevated levels of cytokines (interleukin-6, tumor necrosis factor alpha, interleukin-1 beta) are frequently observed; these cytokines can cross the blood–brain barrier and influence microglial activation (Paul et al., 2022). Immune cells residing in the brain produce reactive oxygen species and nitric oxide, resulting in neurotoxicity. An important role is also played by tryptophan metabolism—cytokines activate indoleamine-pyrrole 2,3-dioxygenase (IDO) and tryptophan 2,3-dioxygenase (TDO), which promote metabolic pathways leading to the production of kynurenine instead of serotonin. Kynurenine is subsequently converted into the neurotoxic quinolinic acid, which activates N-methyl-D-aspartate (NMDA) receptors, leading to excessive neuronal excitation and neuronal death (Paul et al., 2022).

## Running Code

```{r}
library(dplyr)
library(readr)
library(ggplot2)
setwd("~/Desktop/HTS_project")
data <- read_tsv("GSE260603_normalized_data.tsv")
```

# PCA

```{r}
# expression matrix
expr <- data %>%
  as.data.frame()

rownames(expr) <- expr$Geneid
expr$Geneid <- NULL

expr <- as.matrix(expr)
storage.mode(expr) <- "numeric"
pca <- prcomp(t(expr), center = TRUE, scale. = TRUE)

# variance explained
var_explained <- (pca$sdev^2 / sum(pca$sdev^2)) * 100

# data frame
pca_df <- as.data.frame(pca$x) %>%
  mutate(sample = rownames(.))

# 5 PC1 vs PC2)
ggplot(pca_df, aes(x = PC1, y = PC2)) +
  geom_point(size = 3, alpha = 0.8) +
  labs(
    title = "PCA (normalized expression)",
    subtitle = paste0(
      "PC1: ", round(var_explained[1], 1), "% | ",
      "PC2: ", round(var_explained[2], 1), "%"
    ),
    x = paste0("PC1 (", round(var_explained[1], 1), "%)"),
    y = paste0("PC2 (", round(var_explained[2], 1), "%)")
  ) +
  theme_minimal(base_size = 14)

```

Principal Component Analysis (PCA) was used to look at overall similarities and differences between samples based on gene expression. The first two principal components explain a limited part of the total variation in the data (PC1: 14.3%, PC2: 10.4%), which means that the variability is spread across many factors rather than driven by one strong effect. The PCA plot shows that most samples cluster together and there is no clear separation between groups along PC1 or PC2.

# PCA + Loadings

```{r}
# loadings = gene contributions to PCs
loadings <- as.data.frame(pca$rotation, stringsAsFactors = FALSE)
loadings$Geneid <- rownames(loadings)

# making sure PC columns are numeric
loadings$PC1 <- as.numeric(loadings$PC1)
loadings$PC2 <- as.numeric(loadings$PC2)

#tTop genes for PC1 / PC2 by absolute loading
top_PC1 <- loadings %>%
  dplyr::mutate(absPC1 = abs(PC1)) %>%
  dplyr::arrange(dplyr::desc(absPC1)) %>%
  dplyr::slice_head(n = 50)

top_PC2 <- loadings %>%
  dplyr::mutate(absPC2 = abs(PC2)) %>%
  dplyr::arrange(dplyr::desc(absPC2)) %>%
  dplyr::slice_head(n = 50)

# showing top 10 gene IDs
top_PC1$Geneid[1:10]
top_PC2$Geneid[1:10]
```

# Genes Function

```{r}
library(biomaRt)
genes_PC1 <- top_PC1$Geneid

# connect to Ensembl
ensembl <- useMart(
  biomart = "ensembl",
  dataset = "hsapiens_gene_ensembl"
)

# get annotations
annot_PC1 <- getBM(
  attributes = c(
    "ensembl_gene_id",
    "hgnc_symbol",
    "description",
    "gene_biotype"
  ),
  filters = "ensembl_gene_id",
  values = genes_PC1,
  mart = ensembl
)

PC1_table <- top_PC1 %>%
  dplyr::select(Geneid, PC1) %>%
  dplyr::left_join(
    annot_PC1,
    by = c("Geneid" = "ensembl_gene_id")
  )
```

```{r}
genes_PC2 <- top_PC2$Geneid

annot_PC2 <- getBM(
  attributes = c(
    "ensembl_gene_id",
    "hgnc_symbol",
    "description",
    "gene_biotype"),
  filters = "ensembl_gene_id",
  values = genes_PC2,
  mart = ensembl
)

PC2_table <- top_PC2 %>%
  dplyr::select(Geneid, PC2) %>%
  dplyr::left_join(
    annot_PC2,
    by = c("Geneid" = "ensembl_gene_id")
  )

```

PC1 is mainly driven by genes involved in immune response, vesicle transport, and cellular metabolism, while PC2 reflects variation related to transcriptional regulation, RNA processing, and protein turnover.

# Heatmaps

```{r}
library(pheatmap)

top_genesPC1 <- top_PC1$Geneid[1:30]

mat <- expr[top_genesPC1, , drop = FALSE]

pheatmap(
  mat,
  scale = "row",          # <- kluczowe
  show_rownames = TRUE,
  show_colnames = FALSE
)
```

Heatmap of genes with the highest PC1 loadings highlights opposing expression patterns across samples, consistent with the continuous separation observed along PC1 in the PCA (14.3% variance explained).

```{r}
library(pheatmap)

top_genesPC2 <- top_PC2$Geneid[1:30]

mat <- expr[top_genesPC2, , drop = FALSE]

pheatmap(
  mat,
  scale = "row",          # <- kluczowe
  show_rownames = TRUE,
  show_colnames = FALSE
)
```

Heatmap of top PC2-contributing genes shows weaker and more localized expression differences across samples, consistent with the limited separation observed along PC2 in the PCA (10.4% variance explained).

## Summary

Overall, the PCA indicates that PC1 captures the main biologically relevant axis of transcriptional variability, while PC2 reflects secondary, more heterogeneous expression differences that do not drive a clear separation of samples.

# UMAP

```{r}
# Pakiety
install.packages(c("uwot", "ggplot2", "readr", "dplyr"))  # jeśli nie masz
library(readr)
library(dplyr)
library(ggplot2)
library(uwot)
```

```{r}
library(ggplot2)
expr <- as.data.frame(data)
rownames(expr) <- expr$Geneid
expr$Geneid <- NULL
expr <- as.matrix(expr)
storage.mode(expr) <- "numeric"

gene_var <- apply(expr, 1, var, na.rm = TRUE)
topN <- 2000
expr_hv <- expr[order(gene_var, decreasing = TRUE)[1:topN], ]

# PCA
pca_umap <- prcomp(t(expr_hv), center = TRUE, scale. = TRUE)

# umap
set.seed(123)
umap_emb <- uwot::umap(
  pca_umap$x[, 1:30],
  n_neighbors = 15,
  min_dist = 0.1,
  metric = "cosine"
)

umap_df <- as.data.frame(umap_emb)
colnames(umap_df) <- c("UMAP1", "UMAP2")
umap_df$sample <- rownames(pca_umap$x)

#plotting
plot(
  umap_df$UMAP1,
  umap_df$UMAP2,
  pch = 16,
  col = rgb(70,130,180,120, maxColorValue = 255),
  xlab = "UMAP1",
  ylab = "UMAP2",
  main = "UMAP (top variable genes → PCA → UMAP)"
)
```

UMAP analysis of highly variable genes shows a continuous pattern rather than clear, separate clusters. This means that gene expression differences between PBMC samples change gradually. The observed variation likely reflects differences in immune cell composition and immune activation, rather than distinct subgroups. This is consistent with the study design, where depression subgroups were identified only after combining RNA-seq data with other data types.

## Bibliography

Abbafati, C., Abbas, K. M., Abbasi, M., Abbasifard, M., Abbasi-
Kangevari, M., Abbastabar, H., Abd-Allah, F., Abdelalim, A., Abdollahi,
M., Abdollahpour, I., Abedi, A., Abedi, P., Abegaz, K. H., Abolhassani,
H., Abosetugn, A. E., Aboyans, V., Abrams, E. M., Abreu, L. G., Abrigo,
M. R. M., … Murray, C. J. L. (2020). Global burden of 369 diseases and
injuries in 204 countries and territories, 1990–2019: a systematic analysis
for the Global Burden of Disease Study 2019. The Lancet, 396(10258),
1204–1222. https://doi.org/10.1016/S0140-6736(20)30925-9

Gałecki, P., & Szulc, A. (2023). Psychiatria. Rozpoznania według ICD-11.
(Vol. 1). Edra Urban & Partner.

Paul, E. R., Schwieler, L., Erhardt, S., Boda, S., Trepci, A., Kämpe, R.,
Asratian, A., Holm, L., Yngve, A., Dantzer, R., Heilig, M., Hamilton, J.
P., & Samuelsson, M. (2022). Peripheral and central kynurenine pathway
abnormalities in major depression. Brain, Behavior, and Immunity, 101,
136–145. https://doi.org/10.1016/j.bbi.2022.01.002

Ridley, M., Rao, G., Schilbach, F., & Patel, V. (2020). Poverty, depression,
and anxiety: Causal evidence and mechanisms. In Science (Vol. 370, Issue
6522). American Association for the Advancement of Science.
https://doi.org/10.1126/science.aay0214
