---
title: "PEA"
format: html
editor: visual
---

Before we start, it’s always good practice to clean-up our environment in case we have hidden objects in the background, free up memory, and set our working directory. We also need to load the necessary libraries. Today we will be working with **clusterProfiler()** package, and several other R libraries for data visualisation and management. You might need to install them first using **install.packages() or BiocManager::install().**

```{r}
# Setting up environment ===================================================

# Clean environment
rm(list = ls(all.names = TRUE)) # will clear all objects including hidden objects
gc() # free up memory and report the memory usage
options(max.print = .Machine$integer.max, scipen = 999, stringsAsFactors = F, dplyr.summarise.inform = F) # avoid truncated output in R console and scientific notation


# Loading relevant libraries 
library(tidyverse) # includes ggplot2, for data visualisation. dplyr, for data manipulation.
library(RColorBrewer) # for a colourful plot
library(pheatmap)
library(clusterProfiler) # for PEA analysis
library(enrichplot) # for visualisations
library(ggupset) # for visualisations

## Set relevant paths
list.files()
in_path <- "/home/ihor/Documents/HTS/PEA/data/"
out_path <- "/home/ihor/Documents/HTS/PEA/clusterProfiler"
dir.create(out_path, showWarnings = F, recursive = T)

```

Let’s get started! The first step is of course to load our **differential gene expression results**. If you have your own dataset, just read it into RStudio, and make sure you have a dataframe as I describe above. We can also add a new column, diffexpressed, which indicates whether a gene is UPregulated, DOWNregulated or NOt differentially expressed. I used a **log2fc = 0 and p-adjusted value of 0.05** to define the thresholds, but you might want to tweak it depending on how permissive you want to be (or how many differentially expressed genes you have!).

```{r}
# Read in data ===================================================
list.files(in_path)
df <- read.csv(paste0(in_path, 'severevshealthy_degresults-1.csv'), row.names = 1)

# Annotate according to differential expression
df <- df %>% mutate(diffexpressed = case_when(
  log2fc > 0 & padj < 0.05 ~ 'UP',
  log2fc < 0 & padj < 0.05 ~ 'DOWN',
  padj > 0.05 ~ 'NO'
))
```

Make sure to download the gene sets that are appropriate for your analysis, and edit the **species** if needed!

```{r}
## Prepare pathways or gene sets -----------------------------------------------
# msigdbr R package provides Molecular Signatures Database (MSigDB) gene sets
# Read more here: https://igordot.github.io/msigdbr/articles/msigdbr-intro.html
# msigdbr_collections()
library(msigdbr)
name_of_gene_set <- 'H' # for hallmark gene sets
# Other options:
# 'C1' - positional gene sets
# 'C2' - curated gene sets (includes KEGG, Reactome, BioCarta, etc.)
# 'C3' - regulatory target gene sets  
# 'C4' - computational gene sets
# 'C5' - ontology gene sets (GO terms)
# 'C6' - oncogenic signature gene sets
# 'C7' - immunologic signature gene sets
# 'C8' - cell type signature gene sets
gene_sets_df <- msigdbr(species = 'Homo sapiens', category = name_of_gene_set)
#gene_sets_df <- msigdbr(species = 'Homo sapiens', category = 'C5', subcollection = 'GO:BP')
head(gene_sets_df)
```

```{r}
# Convert to named list format required by clusterprofiler: the first column should have pathway names
# the second column should have gene symbols
gene_sets <- gene_sets_df %>%
  dplyr::select(gs_name, gene_symbol)

cat(paste("Loaded", length(unique(gene_sets$gs_name)), "gene sets\n"))
```

We will:

-   **Remove non-significant genes:** since we only want the enriched pathways of the significantly differentially expressed genes.

-   **Substitute names** so they are annotated nicely in the heatmap later

-   **Split the data** into a list of 2 sub-dataframes: upregulated, downregulated genes.

```{r}
## 2. Prepare deg data -----------------------------------------------

# Annotate according to differential expression
df <- df %>% mutate(diffexpressed = case_when(
  log2fc > 0 & padj < 0.05 ~ 'UP',
  log2fc < 0 & padj < 0.05 ~ 'DOWN',
  padj > 0.05 ~ 'NO'
  
))
# Remove non-significant genes for ORA
df_sig <- df[df$diffexpressed != 'NO', ] # remove non-significant genes
cat(paste("Found", nrow(df_sig), "significantly differentially expressed genes\n"))
cat(paste("UP:", sum(df_sig$diffexpressed == 'UP'), "\n"))
cat(paste("DOWN:", sum(df_sig$diffexpressed == 'DOWN'), "\n"))

# Split the dataframe into a list of sub-dataframes: upregulated, downregulated genes
deg_results_list <- split(df_sig, df_sig$diffexpressed)
```

As I mentioned in my previous blogpost, it’s important to provide a set of **background genes** against which to compare your DGE results. This helps avoid **bias** and get more accurate statistical results.\
For example, if you are studying neural cells, it doesn’t make sense to include muscle-specific genes when deciding whether a pathway is upregulated or not – those genes could have never been expressed in neural cells so they shouldn’t be taken into account.\
Sometimes it’s hard to decide what “could have been” measured in your experiment, but a good approximation is to **include all genes you detected as part of your “gene universe”**.

```{r}
## 4. Prepare background genes ----------------------------------------
# These should be ALL the genes you could have measured in your experiment,
# to avoid bias. I will use all the genes from the original dataset we used
# to compare severe vs healthy cells
data <- scRNAseq::BacherTCellData(filtered = TRUE, ensembl = FALSE, location = TRUE)
background_genes <- row.names(data)
background_genes[1:5]
rm(data) # to clear some space
```

Wohoo! After all these steps… we are finally ready to run clusterProfiler! Are you excited?

First, let’s set some variables. This is not essential but it will make your life easier in case you need to change them later on.

```{r}
## 3. Run ClusterProfiler -----------------------------------------------

# Settings
name_of_comparison <- 'severevshealthy'
padj_cutoff <- 0.05
genecount_cutoff <- 5
filename_prefix <- paste0(name_of_comparison, '_', name_of_gene_set)
```

Now that we have our pathways and our differential expressed genes ready… let’s call the enricher() function from clusterProfiler.

We use lapply as we are applying the function to our list (remember we split the data into a list of 2 subsets).

If you only have 1 dataset (e.g., just upregulated genes results), you can run the enricher function directly:

```         
x <- enricher(my_dge_results, TERM2GENE = my_pathways, universe = 
```

```{r}
# Run clusterProfiler on each sub-dataframe
res <- lapply(names(deg_results_list),
              function(x) enricher(gene = deg_results_list[[x]]$gene_symbol,
                                   TERM2GENE = gene_sets,
                                   universe = background_genes,
                                   # Let's use the cut-offs later to filter ORA results
                                   #pvalueCutoff = padj_cutoff,
                                   #minGSSize = genecount_cutoff,
                                   qvalueCutoff = 0.2))
names(res) <- names(deg_results_list)

# Check results
cat("Enrichment analysis results:\n")
for(i in names(res)) {
  cat(paste(i, ":", nrow(as.data.frame(res[[i]])), "enriched pathways\n"))
}
```

What does the enricher() function do?

It basically does a hypergeometric test for pathway enrichment analysis, accepting a user-defined annotation. So it is very useful if you want to analyse your data with unsupported organisms or other ontologies/pathways, or customised annotations.

Its inputs are the dge results, and two additional parameters `TERM2GENE` and `TERM2NAME`.

-   `TERM2GENE` is a data.frame with first column of term ID and second column of corresponding mapped gene. It is used to match the pathway with the genes.

-   `TERM2NAME` is optional, used to rename pathways. It is a `data.frame` with first column of term ID and second column of corresponding term name.

    Great! We now have a **list** with our pathway enrichment analysis results. Let’s make it more readable and convert it to a **dataframe** with the following lines of code:

```{r}
#Convert the enrichResults to a dataframe with the pathways
res_df <- lapply(names(res), function(x) rbind(res[[x]]@result))
names(res_df) <- names(res)
res_df <- do.call(rbind, res_df)
head(res_df)
```

And maybe **add a few more columns** to our dataframe ):

```{r}
res_df_sig <- res_df %>% mutate(minuslog10padj = -log10(p.adjust))
```

Now, we can **subset our results**. Filter the significant pathways according to **p-adjusted threshold and gene count.**

```{r}
# Subset to those pathways that have p adj < cutoff and gene count > cutoff (you can also do this in the enricher function)
res_df_sig <- res_df_sig %>% 
  dplyr::filter(p.adjust < padj_cutoff & Count > genecount_cutoff) # select only target pathways have p adjusted < 0.05 and at least 6 genes

```

Finally, don’t forget to **save** your results!

```{r}
print('Saving clusterprofiler results')
write.csv(res_df_sig, file.path(out_path, paste0(filename_prefix, '_resclusterp.csv')), row.names = FALSE)

# Also save the enrichresults object for plotting later
saveRDS(res, file.path(out_path,
        paste0(filename_prefix, "_resclusterp_enrichres.rds")))
# You can also save it as RDS

```
